{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import Bio as bio\n",
    "import re\n",
    "import os\n",
    "import glob\n",
    "from Bio import SeqIO\n",
    "import pprint\n",
    "from BioSQL import BioSeqDatabase\n",
    "from operator import add"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-2-b44e7bd49ed3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mgb_file\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34mr'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Data\\Human_RNA\\human_4_rna.gbff'\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mmega_file\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgb_file\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m\"r\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"genbank\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 4\u001b[1;33m \u001b[1;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mmega_file\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      5\u001b[0m     \u001b[0mi\u001b[0m\u001b[1;33m+=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      6\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mi\u001b[0m \u001b[1;33m%\u001b[0m \u001b[1;36m100\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\SeqIO\\Interfaces.py\u001b[0m in \u001b[0;36m__next__\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m     71\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__next__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     72\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 73\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mnext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrecords\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     74\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     75\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshould_close_stream\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\GenBank\\Scanner.py\u001b[0m in \u001b[0;36mparse_records\u001b[1;34m(self, handle, do_features)\u001b[0m\n\u001b[0;32m    514\u001b[0m         \u001b[1;32mwith\u001b[0m \u001b[0mas_handle\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mhandle\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    515\u001b[0m             \u001b[1;32mwhile\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 516\u001b[1;33m                 \u001b[0mrecord\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    517\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0mrecord\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    518\u001b[0m                     \u001b[1;32mbreak\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\GenBank\\Scanner.py\u001b[0m in \u001b[0;36mparse\u001b[1;34m(self, handle, do_features)\u001b[0m\n\u001b[0;32m    497\u001b[0m         )\n\u001b[0;32m    498\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 499\u001b[1;33m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfeed\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    500\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mconsumer\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    501\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\GenBank\\Scanner.py\u001b[0m in \u001b[0;36mfeed\u001b[1;34m(self, handle, consumer, do_features)\u001b[0m\n\u001b[0;32m    468\u001b[0m         \u001b[1;31m# Features (common to both EMBL and GenBank):\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    469\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mdo_features\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 470\u001b[1;33m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_feed_feature_table\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mconsumer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse_features\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mskip\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    471\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    472\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse_features\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mskip\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# ignore the data\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\Bio\\GenBank\\Scanner.py\u001b[0m in \u001b[0;36mparse_features\u001b[1;34m(self, skip)\u001b[0m\n\u001b[0;32m    227\u001b[0m                     \u001b[1;31m# white space (e.g. out of spec files with too much indentation)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    228\u001b[0m                     \u001b[0mfeature_lines\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mline\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mFEATURE_QUALIFIER_INDENT\u001b[0m \u001b[1;33m:\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 229\u001b[1;33m                     \u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mhandle\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    230\u001b[0m                 \u001b[0mfeatures\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse_feature\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfeature_key\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfeature_lines\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    231\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mline\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\encodings\\cp1252.py\u001b[0m in \u001b[0;36mdecode\u001b[1;34m(self, input, final)\u001b[0m\n\u001b[0;32m     20\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     21\u001b[0m \u001b[1;32mclass\u001b[0m \u001b[0mIncrementalDecoder\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcodecs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mIncrementalDecoder\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 22\u001b[1;33m     \u001b[1;32mdef\u001b[0m \u001b[0mdecode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfinal\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     23\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mcodecs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcharmap_decode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0minput\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0merrors\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mdecoding_table\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     24\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "i = 0\n",
    "gb_file = r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Data\\Human_RNA\\human_4_rna.gbff'\n",
    "mega_file = SeqIO.parse(open(gb_file,\"r\"), \"genbank\")\n",
    "for record in mega_file:\n",
    "    i+=1\n",
    "    if i % 100 == 0:\n",
    "        print(i)\n",
    "print(i)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(3375), strand=1), type='source'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(3375), strand=1), type='gene'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(1102), strand=1), type='exon'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(0), ExactPosition(1), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(8), ExactPosition(16), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(11), ExactPosition(12), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(16), ExactPosition(17), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(19), ExactPosition(20), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(20), ExactPosition(21), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(22), ExactPosition(23), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(23), ExactPosition(24), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(24), ExactPosition(25), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(24), ExactPosition(25), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(27), ExactPosition(28), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(30), ExactPosition(31), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(33), ExactPosition(34), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(34), ExactPosition(35), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(35), ExactPosition(36), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(38), ExactPosition(39), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(47), ExactPosition(48), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(55), ExactPosition(56), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(58), ExactPosition(59), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(59), ExactPosition(60), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(65), ExactPosition(66), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(77), ExactPosition(78), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(84), ExactPosition(85), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(90), ExactPosition(92), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(90), ExactPosition(91), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(91), ExactPosition(92), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(92), ExactPosition(93), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(98), ExactPosition(99), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(102), ExactPosition(103), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(103), ExactPosition(104), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(104), ExactPosition(105), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(107), ExactPosition(108), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(111), ExactPosition(112), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(116), ExactPosition(117), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(118), ExactPosition(119), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(120), ExactPosition(121), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(125), ExactPosition(126), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(134), ExactPosition(135), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(138), ExactPosition(139), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(146), ExactPosition(147), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(147), ExactPosition(148), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(149), ExactPosition(150), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(150), ExactPosition(151), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(151), ExactPosition(152), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(158), ExactPosition(159), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(160), ExactPosition(161), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(162), ExactPosition(163), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(166), ExactPosition(167), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(168), ExactPosition(169), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(172), ExactPosition(173), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(179), ExactPosition(180), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(180), ExactPosition(189), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(180), ExactPosition(181), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(188), ExactPosition(189), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(189), ExactPosition(190), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(191), ExactPosition(192), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(194), ExactPosition(195), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(200), ExactPosition(201), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(219), ExactPosition(220), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(225), ExactPosition(226), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(228), ExactPosition(229), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(229), ExactPosition(230), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(232), ExactPosition(233), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(236), ExactPosition(239), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(237), ExactPosition(238), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(239), ExactPosition(240), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(258), ExactPosition(259), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(261), ExactPosition(262), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(262), ExactPosition(263), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(274), ExactPosition(275), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(275), ExactPosition(276), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(280), ExactPosition(281), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(282), ExactPosition(283), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(288), ExactPosition(289), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(294), ExactPosition(298), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(294), ExactPosition(295), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(307), ExactPosition(308), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(308), ExactPosition(309), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(317), ExactPosition(320), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(327), ExactPosition(328), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(330), ExactPosition(331), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(333), ExactPosition(334), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(336), ExactPosition(338), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(336), ExactPosition(337), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(339), ExactPosition(340), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(351), ExactPosition(352), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(363), ExactPosition(364), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(364), ExactPosition(365), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(367), ExactPosition(368), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(369), ExactPosition(370), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(375), ExactPosition(376), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(376), ExactPosition(377), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(379), ExactPosition(380), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(384), ExactPosition(385), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(389), ExactPosition(390), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(390), ExactPosition(391), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(402), ExactPosition(403), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(407), ExactPosition(411), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(419), ExactPosition(420), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(426), ExactPosition(427), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(430), ExactPosition(431), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(431), ExactPosition(432), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(445), ExactPosition(446), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(448), ExactPosition(449), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(450), ExactPosition(451), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(451), ExactPosition(452), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(455), ExactPosition(456), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(460), ExactPosition(463), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(460), ExactPosition(461), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(462), ExactPosition(463), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(463), ExactPosition(464), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(464), ExactPosition(465), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(465), ExactPosition(466), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(471), ExactPosition(472), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(474), ExactPosition(475), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(475), ExactPosition(476), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(478), ExactPosition(479), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(479), ExactPosition(480), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(484), ExactPosition(487), strand=1), type='misc_feature'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(487), ExactPosition(489), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(490), ExactPosition(491), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(492), ExactPosition(493), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(494), ExactPosition(495), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(495), ExactPosition(496), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(496), ExactPosition(1450), strand=1), type='CDS'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(649), ExactPosition(652), strand=1), type='misc_feature'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(670), ExactPosition(673), strand=1), type='misc_feature'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(910), ExactPosition(913), strand=1), type='misc_feature'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1036), ExactPosition(1039), strand=1), type='misc_feature'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1183), ExactPosition(1429), strand=1), type='misc_feature'),\n",
       " SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1183), ExactPosition(1195), strand=1), FeatureLocation(ExactPosition(1294), ExactPosition(1297), strand=1), FeatureLocation(ExactPosition(1300), ExactPosition(1312), strand=1), FeatureLocation(ExactPosition(1354), ExactPosition(1366), strand=1)], 'order'), type='misc_feature', location_operator='order'),\n",
       " SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1189), ExactPosition(1192), strand=1), FeatureLocation(ExactPosition(1282), ExactPosition(1285), strand=1), FeatureLocation(ExactPosition(1294), ExactPosition(1297), strand=1), FeatureLocation(ExactPosition(1315), ExactPosition(1318), strand=1), FeatureLocation(ExactPosition(1417), ExactPosition(1420), strand=1), FeatureLocation(ExactPosition(1423), ExactPosition(1426), strand=1)], 'order'), type='misc_feature', location_operator='order'),\n",
       " SeqFeature(CompoundLocation([FeatureLocation(ExactPosition(1264), ExactPosition(1267), strand=1), FeatureLocation(ExactPosition(1291), ExactPosition(1294), strand=1)], 'order'), type='misc_feature', location_operator='order'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(496), ExactPosition(497), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(498), ExactPosition(499), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(501), ExactPosition(502), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(502), ExactPosition(518), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(502), ExactPosition(503), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(505), ExactPosition(506), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(507), ExactPosition(508), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(509), ExactPosition(510), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(511), ExactPosition(512), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(514), ExactPosition(515), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(518), ExactPosition(519), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(519), ExactPosition(520), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(520), ExactPosition(521), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(521), ExactPosition(526), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(521), ExactPosition(522), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(523), ExactPosition(524), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(525), ExactPosition(526), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(527), ExactPosition(528), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(528), ExactPosition(529), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(530), ExactPosition(535), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(533), ExactPosition(534), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(538), ExactPosition(539), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(546), ExactPosition(547), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(547), ExactPosition(548), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(548), ExactPosition(549), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(549), ExactPosition(550), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(551), ExactPosition(552), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(554), ExactPosition(555), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(558), ExactPosition(559), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(559), ExactPosition(560), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(560), ExactPosition(561), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(561), ExactPosition(562), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(564), ExactPosition(565), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(571), ExactPosition(572), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(576), ExactPosition(577), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(577), ExactPosition(578), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(578), ExactPosition(579), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(578), ExactPosition(579), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(582), ExactPosition(583), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(583), ExactPosition(584), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(584), ExactPosition(585), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(585), ExactPosition(586), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(586), ExactPosition(587), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(589), ExactPosition(590), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(591), ExactPosition(592), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(593), ExactPosition(594), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(594), ExactPosition(595), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(595), ExactPosition(596), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(596), ExactPosition(597), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(598), ExactPosition(599), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(601), ExactPosition(604), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(601), ExactPosition(602), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(606), ExactPosition(607), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(607), ExactPosition(608), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(608), ExactPosition(609), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(609), ExactPosition(610), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(610), ExactPosition(611), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(614), ExactPosition(615), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(617), ExactPosition(618), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(620), ExactPosition(621), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(622), ExactPosition(623), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(624), ExactPosition(625), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(625), ExactPosition(626), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(626), ExactPosition(627), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(627), ExactPosition(628), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(628), ExactPosition(629), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(629), ExactPosition(630), strand=1), type='misc_feature'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(629), ExactPosition(630), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(631), ExactPosition(635), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(631), ExactPosition(632), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(635), ExactPosition(636), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(637), ExactPosition(638), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(639), ExactPosition(640), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(640), ExactPosition(641), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(642), ExactPosition(643), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(645), ExactPosition(646), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(646), ExactPosition(647), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(648), ExactPosition(649), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(649), ExactPosition(650), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(659), ExactPosition(660), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(660), ExactPosition(661), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(666), ExactPosition(667), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(670), ExactPosition(671), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(671), ExactPosition(672), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(672), ExactPosition(673), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(675), ExactPosition(676), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(676), ExactPosition(677), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(677), ExactPosition(678), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(679), ExactPosition(680), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(682), ExactPosition(683), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(683), ExactPosition(684), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(686), ExactPosition(687), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(687), ExactPosition(688), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(688), ExactPosition(689), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(696), ExactPosition(697), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(702), ExactPosition(703), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(705), ExactPosition(706), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(709), ExactPosition(710), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(710), ExactPosition(711), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(711), ExactPosition(712), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(712), ExactPosition(713), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(717), ExactPosition(718), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(718), ExactPosition(719), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(720), ExactPosition(721), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(723), ExactPosition(724), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(725), ExactPosition(726), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(726), ExactPosition(727), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(729), ExactPosition(730), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(733), ExactPosition(734), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(743), ExactPosition(744), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(744), ExactPosition(745), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(747), ExactPosition(748), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(748), ExactPosition(749), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(750), ExactPosition(751), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(755), ExactPosition(756), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(757), ExactPosition(758), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(759), ExactPosition(760), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(760), ExactPosition(761), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(763), ExactPosition(764), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(764), ExactPosition(765), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(765), ExactPosition(766), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(766), ExactPosition(767), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(771), ExactPosition(772), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(774), ExactPosition(775), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(777), ExactPosition(778), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(778), ExactPosition(779), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(779), ExactPosition(780), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(781), ExactPosition(782), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(783), ExactPosition(784), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(784), ExactPosition(785), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(789), ExactPosition(790), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(793), ExactPosition(794), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(794), ExactPosition(795), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(795), ExactPosition(796), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(799), ExactPosition(800), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(802), ExactPosition(803), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(804), ExactPosition(805), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(806), ExactPosition(807), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(810), ExactPosition(811), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(811), ExactPosition(812), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(814), ExactPosition(815), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(816), ExactPosition(817), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(817), ExactPosition(818), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(821), ExactPosition(822), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(823), ExactPosition(824), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(824), ExactPosition(825), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(828), ExactPosition(829), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(829), ExactPosition(830), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(830), ExactPosition(831), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(831), ExactPosition(832), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(832), ExactPosition(833), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(833), ExactPosition(834), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(834), ExactPosition(835), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(835), ExactPosition(836), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(836), ExactPosition(837), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(842), ExactPosition(843), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(845), ExactPosition(846), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(849), ExactPosition(850), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(850), ExactPosition(851), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(852), ExactPosition(853), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(854), ExactPosition(855), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(855), ExactPosition(856), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(856), ExactPosition(857), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(857), ExactPosition(858), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(862), ExactPosition(863), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(864), ExactPosition(865), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(873), ExactPosition(874), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(875), ExactPosition(882), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(882), ExactPosition(883), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(884), ExactPosition(885), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(890), ExactPosition(891), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(891), ExactPosition(892), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(892), ExactPosition(893), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(894), ExactPosition(895), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(895), ExactPosition(896), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(897), ExactPosition(898), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(898), ExactPosition(899), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(900), ExactPosition(901), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(901), ExactPosition(902), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(903), ExactPosition(904), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(905), ExactPosition(906), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(909), ExactPosition(910), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(910), ExactPosition(911), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(911), ExactPosition(912), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(912), ExactPosition(913), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(913), ExactPosition(914), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(914), ExactPosition(915), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(917), ExactPosition(918), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(919), ExactPosition(920), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(920), ExactPosition(921), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(922), ExactPosition(923), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(923), ExactPosition(924), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(926), ExactPosition(927), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(928), ExactPosition(929), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(930), ExactPosition(936), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(930), ExactPosition(931), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(931), ExactPosition(932), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(932), ExactPosition(933), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(934), ExactPosition(935), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(936), ExactPosition(937), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(938), ExactPosition(939), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(940), ExactPosition(941), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(941), ExactPosition(942), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(943), ExactPosition(944), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(945), ExactPosition(946), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(946), ExactPosition(962), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(947), ExactPosition(948), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(949), ExactPosition(950), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(950), ExactPosition(951), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(952), ExactPosition(953), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(955), ExactPosition(956), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(956), ExactPosition(957), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(957), ExactPosition(958), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(958), ExactPosition(964), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(958), ExactPosition(959), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(960), ExactPosition(961), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(963), ExactPosition(964), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(964), ExactPosition(965), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(971), ExactPosition(972), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(972), ExactPosition(973), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(974), ExactPosition(975), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(975), ExactPosition(976), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(977), ExactPosition(978), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(978), ExactPosition(979), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(980), ExactPosition(981), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(981), ExactPosition(982), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(983), ExactPosition(984), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(984), ExactPosition(985), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(986), ExactPosition(987), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(987), ExactPosition(988), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(988), ExactPosition(989), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(989), ExactPosition(990), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(990), ExactPosition(991), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(993), ExactPosition(994), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(996), ExactPosition(997), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(998), ExactPosition(999), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(999), ExactPosition(1000), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1000), ExactPosition(1001), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1001), ExactPosition(1002), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1004), ExactPosition(1005), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1005), ExactPosition(1006), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1007), ExactPosition(1008), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1008), ExactPosition(1009), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1010), ExactPosition(1011), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1011), ExactPosition(1012), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1012), ExactPosition(1013), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1014), ExactPosition(1015), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1015), ExactPosition(1016), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1016), ExactPosition(1017), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1017), ExactPosition(1018), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1018), ExactPosition(1019), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1020), ExactPosition(1021), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1022), ExactPosition(1023), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1026), ExactPosition(1027), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1027), ExactPosition(1028), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1029), ExactPosition(1030), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1030), ExactPosition(1031), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1032), ExactPosition(1033), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1033), ExactPosition(1034), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1034), ExactPosition(1035), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1035), ExactPosition(1036), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1044), ExactPosition(1045), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1048), ExactPosition(1049), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1050), ExactPosition(1051), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1054), ExactPosition(1055), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1061), ExactPosition(1062), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1062), ExactPosition(1063), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1070), ExactPosition(1071), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1071), ExactPosition(1072), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1072), ExactPosition(1073), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1077), ExactPosition(1078), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1081), ExactPosition(1082), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1083), ExactPosition(1084), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1084), ExactPosition(1085), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1085), ExactPosition(1086), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1090), ExactPosition(1091), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1094), ExactPosition(1095), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1102), ExactPosition(1154), strand=1), type='exon'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1102), ExactPosition(1103), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1116), ExactPosition(1117), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1117), ExactPosition(1118), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1121), ExactPosition(1122), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1123), ExactPosition(1124), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1127), ExactPosition(1128), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1128), ExactPosition(1137), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1135), ExactPosition(1136), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1136), ExactPosition(1137), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1142), ExactPosition(1143), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1147), ExactPosition(1148), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1149), ExactPosition(1150), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1150), ExactPosition(1151), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1154), ExactPosition(1351), strand=1), type='exon'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1154), ExactPosition(1155), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1156), ExactPosition(1157), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1161), ExactPosition(1162), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1162), ExactPosition(1163), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1170), ExactPosition(1171), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1171), ExactPosition(1172), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1176), ExactPosition(1177), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1180), ExactPosition(1181), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1181), ExactPosition(1182), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1188), ExactPosition(1189), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1193), ExactPosition(1194), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1197), ExactPosition(1198), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1198), ExactPosition(1199), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1200), ExactPosition(1201), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1203), ExactPosition(1204), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1205), ExactPosition(1206), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1210), ExactPosition(1211), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1215), ExactPosition(1216), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1218), ExactPosition(1219), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1230), ExactPosition(1231), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1231), ExactPosition(1232), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1232), ExactPosition(1233), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1233), ExactPosition(1234), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1240), ExactPosition(1241), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1242), ExactPosition(1243), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1243), ExactPosition(1244), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1245), ExactPosition(1246), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1247), ExactPosition(1248), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1248), ExactPosition(1249), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1249), ExactPosition(1250), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1251), ExactPosition(1252), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1254), ExactPosition(1255), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1255), ExactPosition(1256), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1260), ExactPosition(1261), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1267), ExactPosition(1268), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1270), ExactPosition(1271), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1275), ExactPosition(1276), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1276), ExactPosition(1277), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1277), ExactPosition(1278), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1280), ExactPosition(1281), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1281), ExactPosition(1282), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1284), ExactPosition(1285), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1285), ExactPosition(1286), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1286), ExactPosition(1287), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1287), ExactPosition(1288), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1292), ExactPosition(1293), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1297), ExactPosition(1298), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1302), ExactPosition(1303), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1303), ExactPosition(1304), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1305), ExactPosition(1306), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1307), ExactPosition(1308), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1311), ExactPosition(1312), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1313), ExactPosition(1314), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1317), ExactPosition(1318), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1320), ExactPosition(1321), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1324), ExactPosition(1325), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1325), ExactPosition(1326), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1327), ExactPosition(1328), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1329), ExactPosition(1330), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1335), ExactPosition(1336), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1338), ExactPosition(1339), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1339), ExactPosition(1340), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1340), ExactPosition(1341), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1342), ExactPosition(1343), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1348), ExactPosition(1349), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1351), ExactPosition(1428), strand=1), type='exon'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1353), ExactPosition(1354), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1357), ExactPosition(1358), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1358), ExactPosition(1359), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1359), ExactPosition(1360), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1360), ExactPosition(1361), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1373), ExactPosition(1374), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1375), ExactPosition(1376), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1387), ExactPosition(1388), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1389), ExactPosition(1390), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1391), ExactPosition(1392), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1402), ExactPosition(1403), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1404), ExactPosition(1405), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1406), ExactPosition(1407), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1407), ExactPosition(1408), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1410), ExactPosition(1411), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1411), ExactPosition(1412), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1412), ExactPosition(1413), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1415), ExactPosition(1416), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1426), ExactPosition(1427), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1427), ExactPosition(1428), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1428), ExactPosition(3375), strand=1), type='exon'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1433), ExactPosition(1434), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1434), ExactPosition(1435), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1436), ExactPosition(1437), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1438), ExactPosition(1439), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1439), ExactPosition(1440), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1440), ExactPosition(1441), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1443), ExactPosition(1444), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1444), ExactPosition(1445), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1445), ExactPosition(1446), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1452), ExactPosition(1453), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1453), ExactPosition(1454), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1459), ExactPosition(1460), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1460), ExactPosition(1461), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1461), ExactPosition(1463), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1461), ExactPosition(1462), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1466), ExactPosition(1467), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1469), ExactPosition(1470), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1471), ExactPosition(1472), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1472), ExactPosition(1475), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1472), ExactPosition(1473), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1473), ExactPosition(1474), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1474), ExactPosition(1475), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1476), ExactPosition(1477), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1478), ExactPosition(1481), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1484), ExactPosition(1485), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1485), ExactPosition(1486), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1488), ExactPosition(1489), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1494), ExactPosition(1495), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1495), ExactPosition(1496), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1496), ExactPosition(1497), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1497), ExactPosition(1498), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1499), ExactPosition(1500), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1500), ExactPosition(1501), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1504), ExactPosition(1505), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1506), ExactPosition(1507), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1507), ExactPosition(1508), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1516), ExactPosition(1517), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1519), ExactPosition(1520), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1520), ExactPosition(1521), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1523), ExactPosition(1524), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1524), ExactPosition(1525), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1527), ExactPosition(1528), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1528), ExactPosition(1529), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1531), ExactPosition(1532), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1532), ExactPosition(1533), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1537), ExactPosition(1538), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1538), ExactPosition(1539), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1540), ExactPosition(1541), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1541), ExactPosition(1542), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1542), ExactPosition(1543), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1544), ExactPosition(1545), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1545), ExactPosition(1546), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1547), ExactPosition(1548), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1548), ExactPosition(1549), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1553), ExactPosition(1554), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1555), ExactPosition(1556), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1558), ExactPosition(1559), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1563), ExactPosition(1589), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1563), ExactPosition(1564), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1564), ExactPosition(1565), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1565), ExactPosition(1566), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1567), ExactPosition(1568), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1568), ExactPosition(1569), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1572), ExactPosition(1573), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1575), ExactPosition(1576), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1578), ExactPosition(1579), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1588), ExactPosition(1589), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1598), ExactPosition(1599), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1606), ExactPosition(1607), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1607), ExactPosition(1608), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1608), ExactPosition(1609), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1615), ExactPosition(1619), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1618), ExactPosition(1619), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1619), ExactPosition(1620), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1620), ExactPosition(1621), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1621), ExactPosition(1622), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1622), ExactPosition(1623), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1623), ExactPosition(1624), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1628), ExactPosition(1629), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1631), ExactPosition(1632), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1638), ExactPosition(1639), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1639), ExactPosition(1640), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1641), ExactPosition(1642), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1645), ExactPosition(1646), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1646), ExactPosition(1647), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1650), ExactPosition(1651), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1651), ExactPosition(1652), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1653), ExactPosition(1654), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1654), ExactPosition(1655), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1656), ExactPosition(1657), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1657), ExactPosition(1658), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1660), ExactPosition(1661), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1666), ExactPosition(1667), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1668), ExactPosition(1669), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1669), ExactPosition(1670), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1671), ExactPosition(1672), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1672), ExactPosition(1673), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1673), ExactPosition(1674), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1681), ExactPosition(1682), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1686), ExactPosition(1687), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1687), ExactPosition(1688), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1692), ExactPosition(1693), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1695), ExactPosition(1696), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1697), ExactPosition(1698), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1701), ExactPosition(1702), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1708), ExactPosition(1709), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1709), ExactPosition(1710), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1711), ExactPosition(1712), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1714), ExactPosition(1715), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1717), ExactPosition(1718), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1720), ExactPosition(1721), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1721), ExactPosition(1722), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1722), ExactPosition(1723), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1723), ExactPosition(1724), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1737), ExactPosition(1738), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1739), ExactPosition(1740), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1740), ExactPosition(1741), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1745), ExactPosition(1746), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1746), ExactPosition(1747), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1748), ExactPosition(1749), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1750), ExactPosition(1751), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1753), ExactPosition(1754), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1755), ExactPosition(1756), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1756), ExactPosition(1757), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1757), ExactPosition(1758), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1764), ExactPosition(1765), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1771), ExactPosition(1772), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1775), ExactPosition(1776), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1777), ExactPosition(1778), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1781), ExactPosition(1782), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1783), ExactPosition(1784), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1784), ExactPosition(1785), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1793), ExactPosition(1794), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1804), ExactPosition(1805), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1807), ExactPosition(1808), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1808), ExactPosition(1823), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1809), ExactPosition(1810), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1814), ExactPosition(1832), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1815), ExactPosition(1816), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1821), ExactPosition(1822), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1822), ExactPosition(1823), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1823), ExactPosition(1835), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1823), ExactPosition(1824), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1824), ExactPosition(1825), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1832), ExactPosition(1833), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1835), ExactPosition(1836), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1846), ExactPosition(1847), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1849), ExactPosition(1852), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1855), ExactPosition(1856), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1862), ExactPosition(1863), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1864), ExactPosition(1865), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1866), ExactPosition(1867), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1872), ExactPosition(1873), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1873), ExactPosition(1877), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1875), ExactPosition(1876), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1877), ExactPosition(1878), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1878), ExactPosition(1885), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1884), ExactPosition(1885), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1895), ExactPosition(1896), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1895), ExactPosition(1896), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1896), ExactPosition(1897), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1899), ExactPosition(1900), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1909), ExactPosition(1910), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1911), ExactPosition(1912), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1932), ExactPosition(1933), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1934), ExactPosition(1935), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1938), ExactPosition(1939), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1942), ExactPosition(1943), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1948), ExactPosition(1949), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1954), ExactPosition(1955), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1958), ExactPosition(1959), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1960), ExactPosition(1961), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1972), ExactPosition(1973), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(1975), ExactPosition(1976), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2000), ExactPosition(2001), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2006), ExactPosition(2007), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2008), ExactPosition(2010), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2013), ExactPosition(2014), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2016), ExactPosition(2017), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2017), ExactPosition(2025), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2026), ExactPosition(2027), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2033), ExactPosition(2034), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2041), ExactPosition(2042), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2043), ExactPosition(2044), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2049), ExactPosition(2050), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2058), ExactPosition(2059), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2068), ExactPosition(2069), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2075), ExactPosition(2076), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2076), ExactPosition(2077), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2081), ExactPosition(2082), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2082), ExactPosition(2083), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2092), ExactPosition(2093), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2093), ExactPosition(2094), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2095), ExactPosition(2096), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2098), ExactPosition(2099), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2116), ExactPosition(2117), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2132), ExactPosition(2133), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2134), ExactPosition(2135), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2135), ExactPosition(2136), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2142), ExactPosition(2143), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2144), ExactPosition(2145), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2155), ExactPosition(2156), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2160), ExactPosition(2161), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2161), ExactPosition(2162), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2162), ExactPosition(2163), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2164), ExactPosition(2165), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2167), ExactPosition(2168), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2175), ExactPosition(2184), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2178), ExactPosition(2179), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2178), ExactPosition(2179), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2180), ExactPosition(2181), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2184), ExactPosition(2185), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2199), ExactPosition(2200), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2201), ExactPosition(2202), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2202), ExactPosition(2203), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2203), ExactPosition(2204), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2210), ExactPosition(2211), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2211), ExactPosition(2212), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2227), ExactPosition(2228), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2235), ExactPosition(2236), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2238), ExactPosition(2239), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2240), ExactPosition(2241), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2248), ExactPosition(2249), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2258), ExactPosition(2259), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2268), ExactPosition(2269), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2271), ExactPosition(2272), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2272), ExactPosition(2273), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2273), ExactPosition(2274), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2274), ExactPosition(2275), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2280), ExactPosition(2281), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2284), ExactPosition(2285), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2285), ExactPosition(2286), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2287), ExactPosition(2288), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2291), ExactPosition(2292), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2294), ExactPosition(2295), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2301), ExactPosition(2302), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2312), ExactPosition(2313), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2313), ExactPosition(2314), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2318), ExactPosition(2319), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2320), ExactPosition(2321), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2321), ExactPosition(2322), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2325), ExactPosition(2327), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2327), ExactPosition(2328), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2328), ExactPosition(2329), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2338), ExactPosition(2339), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2340), ExactPosition(2341), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2341), ExactPosition(2342), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2347), ExactPosition(2348), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2352), ExactPosition(2353), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2353), ExactPosition(2354), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2361), ExactPosition(2362), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2362), ExactPosition(2363), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2365), ExactPosition(2366), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2368), ExactPosition(2370), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2371), ExactPosition(2372), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2379), ExactPosition(2380), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2380), ExactPosition(2381), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2382), ExactPosition(2383), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2383), ExactPosition(2384), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2388), ExactPosition(2389), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2391), ExactPosition(2392), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2397), ExactPosition(2398), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2398), ExactPosition(2401), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2398), ExactPosition(2399), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2399), ExactPosition(2400), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2401), ExactPosition(2402), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2408), ExactPosition(2409), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2416), ExactPosition(2417), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2417), ExactPosition(2418), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2426), ExactPosition(2427), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2428), ExactPosition(2429), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2442), ExactPosition(2443), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2443), ExactPosition(2444), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2445), ExactPosition(2446), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2447), ExactPosition(2448), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2457), ExactPosition(2458), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2464), ExactPosition(2465), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2465), ExactPosition(2466), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2466), ExactPosition(2467), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2474), ExactPosition(2475), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2475), ExactPosition(2476), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2481), ExactPosition(2482), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2484), ExactPosition(2485), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2492), ExactPosition(2493), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2493), ExactPosition(2494), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2494), ExactPosition(2495), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2498), ExactPosition(2499), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2505), ExactPosition(2506), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2506), ExactPosition(2507), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2508), ExactPosition(2509), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2514), ExactPosition(2515), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2515), ExactPosition(2524), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2515), ExactPosition(2516), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2517), ExactPosition(2518), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2524), ExactPosition(2525), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2527), ExactPosition(2528), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2528), ExactPosition(2529), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2531), ExactPosition(2532), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2533), ExactPosition(2534), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2539), ExactPosition(2540), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2542), ExactPosition(2543), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2545), ExactPosition(2557), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2546), ExactPosition(2547), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2549), ExactPosition(2550), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2550), ExactPosition(2551), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2561), ExactPosition(2562), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2565), ExactPosition(2566), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2566), ExactPosition(2567), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2573), ExactPosition(2574), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2579), ExactPosition(2580), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2584), ExactPosition(2585), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2588), ExactPosition(2589), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2594), ExactPosition(2595), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2599), ExactPosition(2600), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2601), ExactPosition(2602), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2608), ExactPosition(2609), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2609), ExactPosition(2610), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2610), ExactPosition(2611), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2613), ExactPosition(2614), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2623), ExactPosition(2624), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2630), ExactPosition(2631), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2631), ExactPosition(2632), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2638), ExactPosition(2640), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2656), ExactPosition(2657), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2661), ExactPosition(2662), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2669), ExactPosition(2670), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2671), ExactPosition(2672), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2674), ExactPosition(2675), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2675), ExactPosition(2676), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2693), ExactPosition(2694), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2698), ExactPosition(2699), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2714), ExactPosition(2715), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2715), ExactPosition(2716), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2717), ExactPosition(2728), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2720), ExactPosition(2721), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2722), ExactPosition(2730), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2727), ExactPosition(2728), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2728), ExactPosition(2731), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2738), ExactPosition(2739), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2739), ExactPosition(2747), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2747), ExactPosition(2748), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2748), ExactPosition(2749), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2749), ExactPosition(2750), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2751), ExactPosition(2752), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2754), ExactPosition(2755), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2757), ExactPosition(2758), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2758), ExactPosition(2759), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2760), ExactPosition(2761), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2776), ExactPosition(2777), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2794), ExactPosition(2795), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2798), ExactPosition(2799), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2799), ExactPosition(2800), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2807), ExactPosition(2808), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2809), ExactPosition(2810), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2816), ExactPosition(2817), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2824), ExactPosition(2825), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2825), ExactPosition(2826), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2836), ExactPosition(2837), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2844), ExactPosition(2845), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2852), ExactPosition(2853), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2855), ExactPosition(2865), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2860), ExactPosition(2861), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2862), ExactPosition(2863), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2863), ExactPosition(2864), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2865), ExactPosition(2866), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2866), ExactPosition(2867), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2871), ExactPosition(2872), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2872), ExactPosition(2873), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2877), ExactPosition(2883), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2879), ExactPosition(2880), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2880), ExactPosition(2893), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2887), ExactPosition(2888), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2890), ExactPosition(2891), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2927), ExactPosition(2928), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2932), ExactPosition(2933), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2934), ExactPosition(2935), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2937), ExactPosition(2938), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2938), ExactPosition(2939), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2943), ExactPosition(2944), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2948), ExactPosition(2949), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2949), ExactPosition(2950), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2954), ExactPosition(2955), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2955), ExactPosition(2956), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2958), ExactPosition(2959), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2959), ExactPosition(2960), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2961), ExactPosition(2962), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2964), ExactPosition(2965), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2967), ExactPosition(2968), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2969), ExactPosition(2970), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2976), ExactPosition(2977), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2980), ExactPosition(2981), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2983), ExactPosition(2984), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2988), ExactPosition(2989), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2989), ExactPosition(2990), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(2995), ExactPosition(2996), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3000), ExactPosition(3001), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3002), ExactPosition(3003), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3022), ExactPosition(3023), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3024), ExactPosition(3025), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3027), ExactPosition(3028), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3029), ExactPosition(3062), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3030), ExactPosition(3031), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3031), ExactPosition(3048), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3032), ExactPosition(3033), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3040), ExactPosition(3041), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3043), ExactPosition(3044), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3045), ExactPosition(3046), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3046), ExactPosition(3047), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3048), ExactPosition(3049), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3050), ExactPosition(3051), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3060), ExactPosition(3061), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3062), ExactPosition(3063), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3063), ExactPosition(3064), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3066), ExactPosition(3067), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3071), ExactPosition(3072), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3078), ExactPosition(3079), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3085), ExactPosition(3086), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3090), ExactPosition(3093), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3094), ExactPosition(3100), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3101), ExactPosition(3102), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3102), ExactPosition(3103), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3108), ExactPosition(3109), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3112), ExactPosition(3116), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3114), ExactPosition(3119), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3116), ExactPosition(3117), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3119), ExactPosition(3120), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3124), ExactPosition(3130), strand=1), type='regulatory'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3124), ExactPosition(3125), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3137), ExactPosition(3138), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3139), ExactPosition(3140), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3141), ExactPosition(3142), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3146), ExactPosition(3147), strand=1), type='polyA_site'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3147), ExactPosition(3155), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3147), ExactPosition(3148), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3161), ExactPosition(3162), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3173), ExactPosition(3178), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3176), ExactPosition(3177), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3178), ExactPosition(3183), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3179), ExactPosition(3180), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3181), ExactPosition(3182), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3198), ExactPosition(3199), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3201), ExactPosition(3202), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3206), ExactPosition(3207), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3208), ExactPosition(3209), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3209), ExactPosition(3210), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3218), ExactPosition(3219), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3228), ExactPosition(3229), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3239), ExactPosition(3240), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3241), ExactPosition(3242), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3242), ExactPosition(3243), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3244), ExactPosition(3245), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3244), ExactPosition(3245), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3250), ExactPosition(3251), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3256), ExactPosition(3257), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3266), ExactPosition(3267), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3267), ExactPosition(3268), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3276), ExactPosition(3277), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3281), ExactPosition(3282), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3287), ExactPosition(3288), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3297), ExactPosition(3303), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3301), ExactPosition(3302), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3302), ExactPosition(3303), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3303), ExactPosition(3304), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3320), ExactPosition(3321), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3327), ExactPosition(3339), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3327), ExactPosition(3328), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3328), ExactPosition(3329), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3329), ExactPosition(3330), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3331), ExactPosition(3332), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3332), ExactPosition(3333), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3333), ExactPosition(3334), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3334), ExactPosition(3335), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3337), ExactPosition(3338), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3339), ExactPosition(3340), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3346), ExactPosition(3347), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3348), ExactPosition(3354), strand=1), type='regulatory'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3353), ExactPosition(3354), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3366), ExactPosition(3367), strand=1), type='variation'),\n",
       " SeqFeature(FeatureLocation(ExactPosition(3374), ExactPosition(3375), strand=1), type='polyA_site')]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Bio.SeqIO.InsdcIO.GenBankIterator at 0x17d814908e0>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mega_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SeqRecord(seq=Seq('GCGGAGGCTTGGGGCAGCCGGGTAGCTCGGAGGTCGTGGCGCTGGGGGCTAGCA...AAA'), id='NM_001171622.2', name='NM_001171622', description='Homo sapiens vascular endothelial growth factor A (VEGFA), transcript variant 8, mRNA', dbxrefs=[])"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'variation'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "record.features[8].type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "127\n",
      "cds\n"
     ]
    }
   ],
   "source": [
    "i = 0\n",
    "while i < len(record.features):\n",
    "    if record.features[i].type == 'CDS':\n",
    "        print(i)\n",
    "        print('cds')\n",
    "        i+= 1\n",
    "    i+=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "994"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(record.features)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Code for extracting gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene\n",
      "G\n",
      "<class 'str'>\n"
     ]
    }
   ],
   "source": [
    "x = record.features[8]\n",
    "\n",
    "y = str(x)\n",
    "z = re.search('location:(.+?)]',y)\n",
    "lo = z.group(1)\n",
    "lo = lo[lo.find('['):]\n",
    "if ' ' in lo:\n",
    "    lo = lo.replace(' ','')\n",
    "if '[' in lo:\n",
    "    loc = lo.replace('[','')\n",
    "mid = loc.find(':')\n",
    "loc1 = int(loc[:mid])\n",
    "loc2 = int(loc[mid+1:])\n",
    "gene = str(record.seq[loc1:loc2])\n",
    "print(record.features[1].type)\n",
    "print(gene)\n",
    "print(type(gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GC_per = (gene.count('G')+gene.count('C'))/len(gene)\n",
    "GC_per"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Function gene extraction returns gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def code_rev(data,j):\n",
    "    x = data.features[j]\n",
    "    y = str(x)\n",
    "    z = re.search('location:(.+?)]',y)\n",
    "    lo = z.group(1)\n",
    "    loc = lo[lo.find('['):]\n",
    "    if ' ' in loc:\n",
    "        loc = loc.replace(' ','')\n",
    "    if '[' in loc:\n",
    "        loc = loc.replace('[','')\n",
    "    if '{' in loc:\n",
    "        loc = loc.replace('{','')\n",
    "    mid = loc.find(':')\n",
    "    loc1 = int(loc[:mid])\n",
    "    loc2 = int(loc[mid+1:])\n",
    "    gene = str(data.seq[loc1:loc2])\n",
    "    if data.features[j].strand == -1:\n",
    "        gene = gene[::-1]\n",
    "        i = 0\n",
    "        gene_rev = ''\n",
    "        while i < len(gene):\n",
    "            if gene[i] =='A':\n",
    "                gene_rev +='T'\n",
    "            elif gene[i] == 'T':\n",
    "                gene_rev += 'A'\n",
    "            elif gene[i] == 'G':\n",
    "                gene_rev += 'C'\n",
    "            elif gene[i] =='C':\n",
    "                gene_rev += 'G'\n",
    "            i+=1\n",
    "        del(gene)\n",
    "        gene = gene_rev\n",
    "    return(gene)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'A'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "code_rev(record,10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Codon finder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TGA\n",
      "127\n"
     ]
    }
   ],
   "source": [
    "j = 0\n",
    "while j < len(record.features):\n",
    "    if record.features[j].type == 'CDS':\n",
    "        codon = code_rev(record,j)\n",
    "        print(codon[-3:])\n",
    "        print(j)\n",
    "        j+= 1\n",
    "    j+=1\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-14-c69a1f29909c>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mstop_codon\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mi\u001b[0m\u001b[1;33m=\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[1;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mdata\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mi\u001b[0m \u001b[1;33m<\u001b[0m\u001b[1;36m10\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'data' is not defined"
     ]
    }
   ],
   "source": [
    "stop_codon = []\n",
    "i= 0\n",
    "for record in data:\n",
    "    print(i)\n",
    "    if i <10:\n",
    "        while j < len(record.features):\n",
    "            if record.features[j].type == 'CDS':\n",
    "                codon = code_rev(record,j)\n",
    "                codon_3 = codon[-3:]\n",
    "                stop_codon.append(codon_3)\n",
    "                j+= 1\n",
    "                i+=1\n",
    "            else:\n",
    "                j+=1\n",
    "                i+=1\n",
    "    else:\n",
    "        print(stop_codon)\n",
    "        break\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-15-c5d84736ba45>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdata\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'data' is not defined"
     ]
    }
   ],
   "source": [
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-16-d1188caf282e>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[1;32mfor\u001b[0m \u001b[0mrecord\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mdata\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrecord\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'data' is not defined"
     ]
    }
   ],
   "source": [
    "for record in data:\n",
    "    print(record.seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'GC_3' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-18-3ef75825e241>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      8\u001b[0m \u001b[0mglobal_gene\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcode_rev\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      9\u001b[0m \u001b[0mglobal_GC\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mglobal_gene\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'G'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mglobal_gene\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcount\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'C'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m/\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mglobal_gene\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 10\u001b[1;33m \u001b[0mglobal_GC3\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mGC_3\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mglobal_gene\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     11\u001b[0m \u001b[0mcodon_four\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAAA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAAT'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAAG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAAC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TGAA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TGAT'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TGAG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TGAC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAGA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAGT'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAGG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'TAGC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'unk'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[1;32mwhile\u001b[0m \u001b[0mj\u001b[0m \u001b[1;33m<\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfeatures\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'GC_3' is not defined"
     ]
    }
   ],
   "source": [
    "j = 0\n",
    "cds_no = 0\n",
    "codon_data = [['TAA',0],['TAG',0],['TGA',0],['unk',0]]\n",
    "codon_GC_data = [['TAA'],['TAG'],['TGA']]\n",
    "codon_GC3_data = [['TAA'],['TAG'],['TGA']]\n",
    "codon_unk =[]\n",
    "codon4_unk = []\n",
    "global_gene = code_rev(data,0)\n",
    "global_GC = (global_gene.count('G')+global_gene.count('C'))/len(global_gene)\n",
    "global_GC3 = GC_3(global_gene)\n",
    "codon_four = [['TAAA',0],['TAAT',0],['TAAG',0],['TAAC',0],['TGAA',0],['TGAT',0],['TGAG',0],['TGAC',0],['TAGA',0],['TAGT',0],['TAGG',0],['TAGC',0],['unk',0]]\n",
    "while j < len(data.features):\n",
    "    try:\n",
    "        if data.features[j].type == 'CDS':\n",
    "            cds_no +=1\n",
    "            gene = code_rev(data,j)\n",
    "            codon = gene[-3:]\n",
    "            GC_per = (gene.count('G')+gene.count('C'))/len(gene)\n",
    "            GC_3per = GC_3(gene)\n",
    "            if codon == 'TAA':\n",
    "                codon_data[0][1] = codon_data[0][1]+1\n",
    "                codon_GC_data[0].append(GC_per)\n",
    "                codon_GC3_data[0].append(GC_3per)\n",
    "                codon_four = four_stop(data,j,codon_four)\n",
    "                j+=1\n",
    "            elif codon == 'TAG':\n",
    "                codon_data[1][1] = codon_data[1][1]+1\n",
    "                codon_GC_data[1].append(GC_per)\n",
    "                codon_GC3_data[1].append(GC_3per)\n",
    "                codon_four = four_stop(data,j,codon_four)\n",
    "                j+=1\n",
    "            elif codon == 'TGA':\n",
    "                codon_data[2][1] = codon_data[2][1]+1\n",
    "                codon_GC_data[2].append(GC_per)\n",
    "                codon_GC3_data[2].append(GC_3per)\n",
    "                codon_four = four_stop(data,j,codon_four)\n",
    "                j+=1\n",
    "            else:\n",
    "                codon_data[3][1] = codon_data[3][1]+1\n",
    "                codon_four = four_stop(data,j,codon_four)\n",
    "                codon_unk.append(codon)\n",
    "                j+=1\n",
    "        else:\n",
    "            j+=1\n",
    "    except:\n",
    "        j+=1\n",
    "codon_df = pd.DataFrame(codon_GC_data, index = ['TAA','TAG','TGA'])\n",
    "print(codon_df.mean(axis=1))   \n",
    "codon_GC3df = pd.DataFrame(codon_GC3_data, index = ['TAA','TAG','TGA'])\n",
    "print(codon_GC3df.mean(axis=1))\n",
    "\n",
    "print(cds_no - (codon_data[0][1]+codon_data[1][1]+codon_data[2][1]))\n",
    "print(codon_data)\n",
    "print(global_GC)\n",
    "print(codon_four)\n",
    "print(global_GC3)\n",
    "print(data.description,codon_unk)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "def GC_3(gene):\n",
    "    i=0\n",
    "    stop = []\n",
    "    GC3_NA = []\n",
    "    while i+3 <= int((len(gene)/3)):\n",
    "        if gene[(i*3+3)]== 'A': \n",
    "            GC3_NA.append('A')\n",
    "            i +=1\n",
    "        elif gene[(i*3+3)]== 'T': \n",
    "            GC3_NA.append('T')\n",
    "            i+=1\n",
    "        elif gene[(i*3+3)]== 'G': \n",
    "            GC3_NA.append('G')\n",
    "            i+=1\n",
    "        elif gene[(i*3+3)]== 'C': \n",
    "            GC3_NA.append('C')\n",
    "            i+=1\n",
    "        else:\n",
    "            i+=1\n",
    "    GC3 = 0\n",
    "    GC3 = ((GC3_NA.count('G') + GC3_NA.count('C'))/(len(gene)/3))    \n",
    "    return(GC3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def four_stop(data,j,codon_four):\n",
    "    x = data.features[j]\n",
    "    y = str(x)\n",
    "    z = re.search('location:(.+?)]',y)\n",
    "    lo = z.group(1)\n",
    "    loc = lo[lo.find('['):]\n",
    "    if ' ' in loc:\n",
    "        loc = loc.replace(' ','')\n",
    "    if '[' in loc:\n",
    "        loc = loc.replace('[','')\n",
    "    if '{' in loc:\n",
    "        loc = loc.replace('{','')\n",
    "    mid = loc.find(':')\n",
    "    if data.features[j].strand ==1: \n",
    "        loc1 = int(loc[:mid])\n",
    "        loc2 = int((loc[mid+1:]))+1\n",
    "        gene = str(data.seq[loc1:loc2])\n",
    "    elif data.features[j].strand == -1:\n",
    "        loc1 = int(loc[:mid])-1\n",
    "        loc2 = int((loc[mid+1:]))\n",
    "        gene = str(data.seq[loc1:loc2])\n",
    "        gene = gene[::-1]\n",
    "        i = 0\n",
    "        gene_rev = ''\n",
    "        while i < len(gene):\n",
    "            if gene[i] =='A':\n",
    "                gene_rev +='T'\n",
    "            elif gene[i] == 'T':\n",
    "                gene_rev += 'A'\n",
    "            elif gene[i] == 'G':\n",
    "                gene_rev += 'C'\n",
    "            elif gene[i] =='C':\n",
    "                gene_rev += 'G'\n",
    "            i+=1\n",
    "        del(gene)\n",
    "        gene = gene_rev\n",
    "    stopper = gene[-4:]\n",
    "    ar = np.array(codon_four)\n",
    "    u = np.where(ar== stopper)\n",
    "    try:\n",
    "        p= [u[0]]\n",
    "        q = int(p[0])\n",
    "        codon_four[q][1] = codon_four[q][1]+1\n",
    "    except:\n",
    "        codon_four[12][1] = codon_four[12][1]+1\n",
    "    return(codon_four)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "def file_runner(data):\n",
    "    j = 0\n",
    "    codon_data = [['TAA',0],['TAG',0],['TGA',0],['unk',0]]\n",
    "    codon_unk =[]\n",
    "    codon_GC_data = [['TAA'],['TAG'],['TGA']]\n",
    "    codon_GC3_data = [['TAA'],['TAG'],['TGA']]\n",
    "    global_gene = code_rev(data,0)\n",
    "    global_GC = (global_gene.count('G')+global_gene.count('C'))/len(global_gene)\n",
    "    global_GC3 = GC_3(global_gene)\n",
    "    codon_four = [['TAAA',0],['TAAT',0],['TAAG',0],['TAAC',0],['TGAA',0],['TGAT',0],['TGAG',0],['TGAC',0],['TAGA',0],['TAGT',0],['TAGG',0],['TAGC',0],['unk',0]]\n",
    "    while j < len(data.features):\n",
    "        try:\n",
    "            if data.features[j].type == 'CDS':\n",
    "                gene = code_rev(data,j)\n",
    "                codon = gene[-3:]\n",
    "                GC_per = (gene.count('G')+gene.count('C'))/len(gene)\n",
    "                GC_3per = GC_3(gene)\n",
    "                if codon == 'TAA':\n",
    "                    codon_data[0][1] = codon_data[0][1]+1\n",
    "                    codon_GC_data[0].append(GC_per)\n",
    "                    codon_GC3_data[0].append(GC_3per)\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                elif codon == 'TAG':\n",
    "                    codon_data[1][1] = codon_data[1][1]+1\n",
    "                    codon_GC_data[1].append(GC_per)\n",
    "                    codon_GC3_data[1].append(GC_3per)\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                elif codon == 'TGA':\n",
    "                    codon_data[2][1] = codon_data[2][1]+1\n",
    "                    codon_GC_data[2].append(GC_per)\n",
    "                    codon_GC3_data[2].append(GC_3per)\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                else:\n",
    "                    codon_data[3][1] = codon_data[3][1]+1\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                    if len(codon_unk) ==0:\n",
    "                        codon_unk.append(data.description)\n",
    "                        codon_unk.append(codon)\n",
    "                    else:\n",
    "                        codon_unk.append(codon)\n",
    "                    \n",
    "            else:\n",
    "                j+=1\n",
    "        except:\n",
    "            j+=1\n",
    "    codon_df = pd.DataFrame(codon_GC_data, index = ['TAA','TAG','TGA'])\n",
    "    codon_GC3df = pd.DataFrame(codon_GC3_data, index = ['TAA','TAG','TGA'])\n",
    "    GC_list = (codon_df.mean(axis = 1))\n",
    "    GC3_list = (codon_GC3df.mean(axis=1))\n",
    "    Name_list = ['Name','TAA','TGA','TAG','Global_GC','Global_GC3','TAA_GC','TGA_GC','TAG_GC','TAA_GC3','TGA_GC3','TAG_GC3','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "    file_data = [data.description,codon_data[0][1],codon_data[2][1],codon_data[1][1],(global_GC),(global_GC3),GC_list[0],GC_list[2],GC_list[1],GC3_list[0],GC3_list[2],GC3_list[1],codon_four[0][1],codon_four[1][1],codon_four[2][1],codon_four[3][1],codon_four[4][1],codon_four[5][1],codon_four[6][1],codon_four[7][1],codon_four[8][1],codon_four[9][1],codon_four[10][1],codon_four[11][1],codon_data[3][1],codon_four[12][1]]\n",
    "    df_file_data = pd.DataFrame([file_data],columns = Name_list)\n",
    "    df_file_data = df_file_data.set_index('Name')\n",
    "    return([df_file_data,codon_unk])\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[                                                    TAA  TGA  TAG  Global_GC  \\\n",
       " Name                                                                           \n",
       " Homo sapiens vascular endothelial growth factor...    0    1    0   0.515556   \n",
       " \n",
       "                                                     Global_GC3  TAA_GC  \\\n",
       " Name                                                                     \n",
       " Homo sapiens vascular endothelial growth factor...    0.567111     NaN   \n",
       " \n",
       "                                                       TGA_GC  TAG_GC  TAA_GC3  \\\n",
       " Name                                                                            \n",
       " Homo sapiens vascular endothelial growth factor...  0.661426     NaN      NaN   \n",
       " \n",
       "                                                     TGA_GC3  ...  TGAA  TGAT  \\\n",
       " Name                                                         ...               \n",
       " Homo sapiens vascular endothelial growth factor...  0.68239  ...     0     0   \n",
       " \n",
       "                                                     TGAG  TGAC  TAGA  TAGT  \\\n",
       " Name                                                                         \n",
       " Homo sapiens vascular endothelial growth factor...     1     0     0     0   \n",
       " \n",
       "                                                     TAGG  TAGC  Unk_Stop  \\\n",
       " Name                                                                       \n",
       " Homo sapiens vascular endothelial growth factor...     0     0         0   \n",
       " \n",
       "                                                     Unk_four  \n",
       " Name                                                          \n",
       " Homo sapiens vascular endothelial growth factor...         0  \n",
       " \n",
       " [1 rows x 25 columns],\n",
       " []]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_runner(record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "Name_list = ['Name','TAA','TGA','TAG','Global_GC','Global_GC3','TAA_GC','TGA_GC','TAG_GC','TAA_GC3','TGA_GC3','TAG_GC3','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TAA, TGA, TAG, Global_GC, Global_GC3, TAA_GC, TGA_GC, TAG_GC, TAA_GC3, TGA_GC3, TAG_GC3, TAAA, TAAT, TAAG, TAAC, TGAA, TGAT, TGAG, TGAC, TAGA, TAGT, TAGG, TAGC, Unk_Stop, Unk_four]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 25 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df = pd.DataFrame(columns = Name_list)\n",
    "archea_df = archea_df.set_index('Name')\n",
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Homo sapiens vascular endothelial growth factor A (VEGFA), transcript variant 8, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.515556</td>\n",
       "      <td>0.567111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.661426</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.68239</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    TAA  TGA  TAG  Global_GC  \\\n",
       "Name                                                                           \n",
       "Homo sapiens vascular endothelial growth factor...    0    1    0   0.515556   \n",
       "\n",
       "                                                    Global_GC3  TAA_GC  \\\n",
       "Name                                                                     \n",
       "Homo sapiens vascular endothelial growth factor...    0.567111     NaN   \n",
       "\n",
       "                                                      TGA_GC  TAG_GC  TAA_GC3  \\\n",
       "Name                                                                            \n",
       "Homo sapiens vascular endothelial growth factor...  0.661426     NaN      NaN   \n",
       "\n",
       "                                                    TGA_GC3  ...  TGAA  TGAT  \\\n",
       "Name                                                         ...               \n",
       "Homo sapiens vascular endothelial growth factor...  0.68239  ...     0     0   \n",
       "\n",
       "                                                    TGAG  TGAC  TAGA  TAGT  \\\n",
       "Name                                                                         \n",
       "Homo sapiens vascular endothelial growth factor...     1     0     0     0   \n",
       "\n",
       "                                                    TAGG  TAGC  Unk_Stop  \\\n",
       "Name                                                                       \n",
       "Homo sapiens vascular endothelial growth factor...     0     0         0   \n",
       "\n",
       "                                                    Unk_four  \n",
       "Name                                                          \n",
       "Homo sapiens vascular endothelial growth factor...         0  \n",
       "\n",
       "[1 rows x 25 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df = pd.DataFrame(columns = Name_list)\n",
    "archea_df = archea_df.set_index('Name')\n",
    "holder_2 = file_runner(data)\n",
    "full_archea_df = holder_2[0]\n",
    "unknown_stops = pd.DataFrame(holder_2[1])\n",
    "full_archea_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adjusted Runner each gene seperate\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "unk_stops = []\n",
    "gb_file = r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Data\\Human_RNA\\human_1_rna.gbff'\n",
    "mega_file = SeqIO.parse(open(gb_file,\"r\"), \"genbank\")\n",
    "for record in mega_file:\n",
    "    if i <10:\n",
    "        holder = file_runner(record)\n",
    "        archea_df = archea_df.append(holder[0])\n",
    "        if holder[1] != []:\n",
    "            unk_stops.append(holder[1])\n",
    "        i+=1\n",
    "    else:\n",
    "        break\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 26, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.584505</td>\n",
       "      <td>0.563593</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.643661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.773006</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 27, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.588255</td>\n",
       "      <td>0.521654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.788703</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 32, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.360846</td>\n",
       "      <td>0.368810</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 38, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.360967</td>\n",
       "      <td>0.369171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens SWI/SNF related, matrix associated, actin dependent regulator of chromatin, subfamily a, member 4 (SMARCA4), transcript variant 8, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.604384</td>\n",
       "      <td>0.427945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.611558</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.632198</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 37, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.373073</td>\n",
       "      <td>0.364379</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens small integral membrane protein 35 (SMIM35), mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.566416</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.557471</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.568966</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens post-GPI attachment to proteins 2 (PGAP2), transcript variant 22, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.543628</td>\n",
       "      <td>0.487759</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens Rho guanine nucleotide exchange factor 10 like (ARHGEF10L), transcript variant 7, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.622620</td>\n",
       "      <td>0.486672</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens dihydrofolate reductase (DHFR), transcript variant 4, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.450828</td>\n",
       "      <td>0.425708</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   TAA TGA TAG  Global_GC  \\\n",
       "Name                                                                        \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.584505   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.588255   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.360846   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.360967   \n",
       "Homo sapiens SWI/SNF related, matrix associated...   0   1   0   0.604384   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.373073   \n",
       "Homo sapiens small integral membrane protein 35...   0   1   0   0.566416   \n",
       "Homo sapiens post-GPI attachment to proteins 2 ...   0   0   0   0.543628   \n",
       "Homo sapiens Rho guanine nucleotide exchange fa...   0   0   0   0.622620   \n",
       "Homo sapiens dihydrofolate reductase (DHFR), tr...   0   0   0   0.450828   \n",
       "\n",
       "                                                    Global_GC3  TAA_GC  \\\n",
       "Name                                                                     \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.563593     NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.521654     NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.368810     NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.369171     NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0.427945     NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.364379     NaN   \n",
       "Homo sapiens small integral membrane protein 35...    0.571429     NaN   \n",
       "Homo sapiens post-GPI attachment to proteins 2 ...    0.487759     NaN   \n",
       "Homo sapiens Rho guanine nucleotide exchange fa...    0.486672     NaN   \n",
       "Homo sapiens dihydrofolate reductase (DHFR), tr...    0.425708     NaN   \n",
       "\n",
       "                                                      TGA_GC  TAG_GC  TAA_GC3  \\\n",
       "Name                                                                            \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.643661     NaN      NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.633194     NaN      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN     NaN      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN     NaN      NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...  0.611558     NaN      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN     NaN      NaN   \n",
       "Homo sapiens small integral membrane protein 35...  0.557471     NaN      NaN   \n",
       "Homo sapiens post-GPI attachment to proteins 2 ...       NaN     NaN      NaN   \n",
       "Homo sapiens Rho guanine nucleotide exchange fa...       NaN     NaN      NaN   \n",
       "Homo sapiens dihydrofolate reductase (DHFR), tr...       NaN     NaN      NaN   \n",
       "\n",
       "                                                     TGA_GC3  ...  TGAA TGAT  \\\n",
       "Name                                                          ...              \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.773006  ...     0    1   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.788703  ...     0    1   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN  ...     0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN  ...     0    0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...  0.632198  ...     0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN  ...     0    0   \n",
       "Homo sapiens small integral membrane protein 35...  0.568966  ...     0    1   \n",
       "Homo sapiens post-GPI attachment to proteins 2 ...       NaN  ...     0    0   \n",
       "Homo sapiens Rho guanine nucleotide exchange fa...       NaN  ...     0    0   \n",
       "Homo sapiens dihydrofolate reductase (DHFR), tr...       NaN  ...     0    0   \n",
       "\n",
       "                                                   TGAG TGAC TAGA TAGT TAGG  \\\n",
       "Name                                                                          \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0    0    0    0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0    0    0    0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    1    0    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0    0    0    0   \n",
       "Homo sapiens small integral membrane protein 35...    0    0    0    0    0   \n",
       "Homo sapiens post-GPI attachment to proteins 2 ...    0    0    0    0    0   \n",
       "Homo sapiens Rho guanine nucleotide exchange fa...    0    0    0    0    0   \n",
       "Homo sapiens dihydrofolate reductase (DHFR), tr...    0    0    0    0    0   \n",
       "\n",
       "                                                   TAGC Unk_Stop Unk_four  \n",
       "Name                                                                       \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0        0        0  \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0        0        0  \n",
       "Homo sapiens transforming acidic coiled-coil co...    0        0        0  \n",
       "Homo sapiens transforming acidic coiled-coil co...    0        0        0  \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0        0        0  \n",
       "Homo sapiens transforming acidic coiled-coil co...    0        0        0  \n",
       "Homo sapiens small integral membrane protein 35...    0        0        0  \n",
       "Homo sapiens post-GPI attachment to proteins 2 ...    0        0        0  \n",
       "Homo sapiens Rho guanine nucleotide exchange fa...    0        0        0  \n",
       "Homo sapiens dihydrofolate reductase (DHFR), tr...    0        0        0  \n",
       "\n",
       "[10 rows x 25 columns]"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TAA, TGA, TAG, Global_GC, Global_GC3, TAA_GC, TGA_GC, TAG_GC, TAA_GC3, TGA_GC3, TAG_GC3, TAAA, TAAT, TAAG, TAAC, TGAA, TGAT, TGAG, TGAC, TAGA, TAGT, TAGG, TAGC, Unk_Stop, Unk_four]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 25 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df = pd.DataFrame(columns = Name_list)\n",
    "archea_df = archea_df.set_index('Name')\n",
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "100\n",
      "200\n",
      "300\n",
      "400\n",
      "500\n",
      "600\n",
      "700\n",
      "800\n",
      "900\n",
      "1000\n",
      "1100\n",
      "1200\n",
      "1300\n",
      "1400\n",
      "1500\n",
      "1600\n",
      "1700\n",
      "1800\n",
      "1900\n",
      "2000\n",
      "2100\n",
      "2200\n",
      "2300\n",
      "2400\n",
      "2500\n",
      "2600\n",
      "2700\n",
      "2800\n",
      "2900\n",
      "3000\n",
      "3100\n",
      "3200\n",
      "3300\n",
      "3400\n",
      "3500\n",
      "3600\n",
      "3700\n",
      "3800\n",
      "3900\n",
      "4000\n",
      "4100\n",
      "4200\n",
      "4300\n",
      "4400\n",
      "4500\n",
      "4600\n",
      "4700\n",
      "4800\n",
      "4900\n",
      "5000\n",
      "5100\n",
      "5200\n",
      "5300\n",
      "5400\n",
      "5500\n",
      "5600\n",
      "5700\n",
      "5800\n",
      "5900\n",
      "6000\n",
      "6100\n",
      "6200\n",
      "6300\n",
      "6400\n",
      "6500\n",
      "6600\n",
      "6700\n",
      "6800\n",
      "6900\n",
      "7000\n",
      "7100\n",
      "7200\n",
      "7300\n",
      "7400\n",
      "7500\n",
      "7600\n",
      "7700\n",
      "7800\n",
      "7900\n",
      "8000\n",
      "8100\n",
      "8200\n",
      "8300\n",
      "8400\n",
      "8500\n",
      "8600\n",
      "8700\n",
      "8800\n",
      "8900\n",
      "9000\n",
      "9100\n",
      "9200\n",
      "9300\n",
      "9400\n",
      "9500\n",
      "9600\n",
      "9700\n",
      "9800\n",
      "9900\n",
      "10000\n",
      "10100\n",
      "10200\n",
      "10300\n",
      "10400\n",
      "10500\n",
      "10600\n",
      "10700\n",
      "10800\n",
      "10900\n",
      "11000\n",
      "11100\n",
      "11200\n",
      "11300\n",
      "11400\n",
      "11500\n",
      "11600\n",
      "11700\n",
      "11800\n",
      "11900\n",
      "12000\n",
      "12100\n",
      "12200\n",
      "12300\n",
      "12400\n",
      "12500\n",
      "12600\n",
      "12700\n",
      "12800\n",
      "12900\n",
      "13000\n",
      "13100\n",
      "13200\n",
      "13300\n",
      "13400\n",
      "13500\n",
      "13600\n",
      "13700\n",
      "13800\n",
      "13900\n",
      "14000\n",
      "14100\n",
      "14200\n",
      "14300\n",
      "14400\n",
      "14500\n",
      "14600\n",
      "14700\n",
      "14800\n",
      "14900\n",
      "15000\n",
      "15100\n",
      "15200\n",
      "15300\n",
      "15400\n",
      "15500\n",
      "15600\n",
      "15700\n",
      "15800\n",
      "15900\n",
      "16000\n",
      "16100\n",
      "16200\n",
      "16300\n",
      "16400\n",
      "16500\n",
      "16600\n",
      "16700\n",
      "16800\n",
      "16900\n",
      "17000\n",
      "17100\n",
      "17200\n",
      "17300\n",
      "17400\n",
      "17500\n",
      "17600\n",
      "17700\n",
      "17800\n",
      "17900\n",
      "18000\n",
      "18100\n",
      "18200\n",
      "18300\n",
      "18400\n",
      "18500\n",
      "18600\n",
      "18700\n",
      "18800\n",
      "18900\n",
      "19000\n",
      "19100\n",
      "19200\n",
      "19300\n",
      "19400\n",
      "19500\n",
      "19600\n",
      "19700\n",
      "19800\n",
      "19900\n"
     ]
    }
   ],
   "source": [
    "i = 0\n",
    "unk_stops = []\n",
    "gb_file = r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Data\\Human_RNA\\human_8_rna.gbff'\n",
    "mega_file = SeqIO.parse(open(gb_file,\"r\"), \"genbank\")\n",
    "for record in mega_file:\n",
    "    if i <10000000:\n",
    "        holder = file_runner(record)\n",
    "        archea_df = archea_df.append(holder[0])\n",
    "        if holder[1] != []:\n",
    "            unk_stops.append(holder[1])\n",
    "        if i % 100 == 0:\n",
    "            print(i)\n",
    "        i+=1\n",
    "    else:\n",
    "        break\n",
    "            \n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "gb_file = r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Data\\Human_RNA\\human_1_rna.gbff'\n",
    "mega_file = SeqIO.parse(open(gb_file,\"r\"), \"genbank\")\n",
    "for record in mega_file:\n",
    "    while i < 10:\n",
    "        while j < len(record.features):\n",
    "            if record.features[j].type == 'CDS':\n",
    "                print(record.features[j])\n",
    "                j+=1\n",
    "            else:\n",
    "                j+=1\n",
    "        i+=1\n",
    "    else:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Bio.SeqIO.InsdcIO.GenBankIterator at 0x28d56812d30>"
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mega_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens leucine rich repeats and guanylate kinase domain containing (LRGUK), transcript variant X9, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.436075</td>\n",
       "      <td>0.450216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.437835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.519293</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens leucine rich repeats and guanylate kinase domain containing (LRGUK), transcript variant X10, misc_RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.436788</td>\n",
       "      <td>0.469430</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens leucine rich repeats and guanylate kinase domain containing (LRGUK), transcript variant X11, misc_RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.445758</td>\n",
       "      <td>0.479692</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens leucine rich repeats and guanylate kinase domain containing (LRGUK), transcript variant X12, misc_RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.438733</td>\n",
       "      <td>0.471963</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens scavenger receptor cysteine rich family member with 4 domains (SSC4D), transcript variant X1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.626819</td>\n",
       "      <td>0.659044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.663361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.720661</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens STX5 divergent transcript (STX5-DT), long non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.557522</td>\n",
       "      <td>0.584071</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP5-AS1), transcript variant 1, long non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.590219</td>\n",
       "      <td>0.632378</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens PRMT5 divergent transcript (PRMT5-DT), long non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.413669</td>\n",
       "      <td>0.410072</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP5-AS1), transcript variant 2, long non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.479915</td>\n",
       "      <td>0.437632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17-AS1), long non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.715997</td>\n",
       "      <td>0.739958</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19922 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   TAA TGA TAG  Global_GC  \\\n",
       "Name                                                                        \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...   0   1   0   0.436075   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...   0   0   0   0.436788   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...   0   0   0   0.445758   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...   0   0   0   0.438733   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...   0   1   0   0.626819   \n",
       "...                                                 ..  ..  ..        ...   \n",
       "Homo sapiens STX5 divergent transcript (STX5-DT...   0   0   0   0.557522   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...   0   0   0   0.590219   \n",
       "Homo sapiens PRMT5 divergent transcript (PRMT5-...   0   0   0   0.413669   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...   0   0   0   0.479915   \n",
       "Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17...   0   0   0   0.715997   \n",
       "\n",
       "                                                    Global_GC3  TAA_GC  \\\n",
       "Name                                                                     \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0.450216     NaN   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0.469430     NaN   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0.479692     NaN   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0.471963     NaN   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0.659044     NaN   \n",
       "...                                                        ...     ...   \n",
       "Homo sapiens STX5 divergent transcript (STX5-DT...    0.584071     NaN   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...    0.632378     NaN   \n",
       "Homo sapiens PRMT5 divergent transcript (PRMT5-...    0.410072     NaN   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...    0.437632     NaN   \n",
       "Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17...    0.739958     NaN   \n",
       "\n",
       "                                                      TGA_GC  TAG_GC  TAA_GC3  \\\n",
       "Name                                                                            \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...  0.437835     NaN      NaN   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...       NaN     NaN      NaN   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...       NaN     NaN      NaN   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...       NaN     NaN      NaN   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...  0.663361     NaN      NaN   \n",
       "...                                                      ...     ...      ...   \n",
       "Homo sapiens STX5 divergent transcript (STX5-DT...       NaN     NaN      NaN   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...       NaN     NaN      NaN   \n",
       "Homo sapiens PRMT5 divergent transcript (PRMT5-...       NaN     NaN      NaN   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...       NaN     NaN      NaN   \n",
       "Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17...       NaN     NaN      NaN   \n",
       "\n",
       "                                                     TGA_GC3  ...  TGAA TGAT  \\\n",
       "Name                                                          ...              \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...  0.519293  ...     0    0   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...       NaN  ...     0    0   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...       NaN  ...     0    0   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...       NaN  ...     0    0   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...  0.720661  ...     0    0   \n",
       "...                                                      ...  ...   ...  ...   \n",
       "Homo sapiens STX5 divergent transcript (STX5-DT...       NaN  ...     0    0   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...       NaN  ...     0    0   \n",
       "Homo sapiens PRMT5 divergent transcript (PRMT5-...       NaN  ...     0    0   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...       NaN  ...     0    0   \n",
       "Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17...       NaN  ...     0    0   \n",
       "\n",
       "                                                   TGAG TGAC TAGA TAGT TAGG  \\\n",
       "Name                                                                          \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0    1    0    0    0   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0    0    0    0    0   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0    0    0    0    0   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0    0    0    0    0   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0    1    0    0    0   \n",
       "...                                                 ...  ...  ...  ...  ...   \n",
       "Homo sapiens STX5 divergent transcript (STX5-DT...    0    0    0    0    0   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...    0    0    0    0    0   \n",
       "Homo sapiens PRMT5 divergent transcript (PRMT5-...    0    0    0    0    0   \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...    0    0    0    0    0   \n",
       "Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17...    0    0    0    0    0   \n",
       "\n",
       "                                                   TAGC Unk_Stop Unk_four  \n",
       "Name                                                                       \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0        0        0  \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0        0        0  \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0        0        0  \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0        0        0  \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0        0        0  \n",
       "...                                                 ...      ...      ...  \n",
       "Homo sapiens STX5 divergent transcript (STX5-DT...    0        0        0  \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...    0        0        0  \n",
       "Homo sapiens PRMT5 divergent transcript (PRMT5-...    0        0        0  \n",
       "Homo sapiens AQP5 and AQP2 antisense RNA 2 (AQP...    0        0        0  \n",
       "Homo sapiens ARHGEF17 antisense RNA 1 (ARHGEF17...    0        0        0  \n",
       "\n",
       "[19922 rows x 25 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unk_stops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "      <th>sum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens leucine rich repeats and guanylate kinase domain containing (LRGUK), transcript variant X9, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.436075</td>\n",
       "      <td>0.450216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.437835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.519293</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens scavenger receptor cysteine rich family member with 4 domains (SSC4D), transcript variant X1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.626819</td>\n",
       "      <td>0.659044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.663361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.720661</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens scavenger receptor cysteine rich family member with 4 domains (SSC4D), transcript variant X2, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.625048</td>\n",
       "      <td>0.580982</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.679771</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.729730</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens chromosome 7 open reading frame 31 (C7orf31), transcript variant X3, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.365942</td>\n",
       "      <td>0.346618</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.407699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.488189</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PREDICTED: Homo sapiens corticotropin releasing hormone receptor 2 (CRHR2), transcript variant X1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.499774</td>\n",
       "      <td>0.463690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.571197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.507282</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens caspase recruitment domain family member 8 (CARD8), transcript variant 18, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.407085</td>\n",
       "      <td>0.415215</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.516204</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens heterogeneous nuclear ribonucleoprotein U (HNRNPU), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.418087</td>\n",
       "      <td>0.386632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.503632</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.607748</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens tumor protein p53 (TP53), transcript variant 2, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.533679</td>\n",
       "      <td>0.569151</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.572770</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.585915</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens sterile alpha motif domain containing 14 (SAMD14), transcript variant 2, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.642452</td>\n",
       "      <td>0.607810</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.657895</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens sterile alpha motif domain containing 14 (SAMD14), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.642047</td>\n",
       "      <td>0.604704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.655456</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12998 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   TAA TGA TAG  Global_GC  \\\n",
       "Name                                                                        \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...   0   1   0   0.436075   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...   0   1   0   0.626819   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...   0   1   0   0.625048   \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...   0   1   0   0.365942   \n",
       "PREDICTED: Homo sapiens corticotropin releasing...   0   1   0   0.499774   \n",
       "...                                                 ..  ..  ..        ...   \n",
       "Homo sapiens caspase recruitment domain family ...   0   0   1   0.407085   \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...   0   1   0   0.418087   \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...   0   1   0   0.533679   \n",
       "Homo sapiens sterile alpha motif domain contain...   0   0   1   0.642452   \n",
       "Homo sapiens sterile alpha motif domain contain...   0   0   1   0.642047   \n",
       "\n",
       "                                                    Global_GC3  TAA_GC  \\\n",
       "Name                                                                     \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0.450216     NaN   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0.659044     NaN   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0.580982     NaN   \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...    0.346618     NaN   \n",
       "PREDICTED: Homo sapiens corticotropin releasing...    0.463690     NaN   \n",
       "...                                                        ...     ...   \n",
       "Homo sapiens caspase recruitment domain family ...    0.415215     NaN   \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...    0.386632     NaN   \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...    0.569151     NaN   \n",
       "Homo sapiens sterile alpha motif domain contain...    0.607810     NaN   \n",
       "Homo sapiens sterile alpha motif domain contain...    0.604704     NaN   \n",
       "\n",
       "                                                      TGA_GC    TAG_GC  \\\n",
       "Name                                                                     \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...  0.437835       NaN   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...  0.663361       NaN   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...  0.679771       NaN   \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...  0.407699       NaN   \n",
       "PREDICTED: Homo sapiens corticotropin releasing...  0.571197       NaN   \n",
       "...                                                      ...       ...   \n",
       "Homo sapiens caspase recruitment domain family ...       NaN  0.516204   \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...  0.503632       NaN   \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...  0.572770       NaN   \n",
       "Homo sapiens sterile alpha motif domain contain...       NaN  0.657895   \n",
       "Homo sapiens sterile alpha motif domain contain...       NaN  0.655456   \n",
       "\n",
       "                                                    TAA_GC3   TGA_GC3  ...  \\\n",
       "Name                                                                   ...   \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...      NaN  0.519293  ...   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...      NaN  0.720661  ...   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...      NaN  0.729730  ...   \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...      NaN  0.488189  ...   \n",
       "PREDICTED: Homo sapiens corticotropin releasing...      NaN  0.507282  ...   \n",
       "...                                                     ...       ...  ...   \n",
       "Homo sapiens caspase recruitment domain family ...      NaN       NaN  ...   \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...      NaN  0.607748  ...   \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...      NaN  0.585915  ...   \n",
       "Homo sapiens sterile alpha motif domain contain...      NaN       NaN  ...   \n",
       "Homo sapiens sterile alpha motif domain contain...      NaN       NaN  ...   \n",
       "\n",
       "                                                    TGAT TGAG TGAC TAGA TAGT  \\\n",
       "Name                                                                           \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...     0    0    1    0    0   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...     0    0    1    0    0   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...     0    0    1    0    0   \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...     0    1    0    0    0   \n",
       "PREDICTED: Homo sapiens corticotropin releasing...     0    0    1    0    0   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "Homo sapiens caspase recruitment domain family ...     0    0    0    0    0   \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...     0    0    0    0    0   \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...     0    0    1    0    0   \n",
       "Homo sapiens sterile alpha motif domain contain...     0    0    0    0    0   \n",
       "Homo sapiens sterile alpha motif domain contain...     0    0    0    0    0   \n",
       "\n",
       "                                                   TAGG TAGC Unk_Stop  \\\n",
       "Name                                                                    \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...    0    0        0   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0    0        0   \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...    0    0        0   \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...    0    0        0   \n",
       "PREDICTED: Homo sapiens corticotropin releasing...    0    0        0   \n",
       "...                                                 ...  ...      ...   \n",
       "Homo sapiens caspase recruitment domain family ...    0    1        0   \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...    0    0        0   \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...    0    0        0   \n",
       "Homo sapiens sterile alpha motif domain contain...    1    0        0   \n",
       "Homo sapiens sterile alpha motif domain contain...    1    0        0   \n",
       "\n",
       "                                                   Unk_four sum  \n",
       "Name                                                             \n",
       "PREDICTED: Homo sapiens leucine rich repeats an...        0   1  \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...        0   1  \n",
       "PREDICTED: Homo sapiens scavenger receptor cyst...        0   1  \n",
       "PREDICTED: Homo sapiens chromosome 7 open readi...        0   1  \n",
       "PREDICTED: Homo sapiens corticotropin releasing...        0   1  \n",
       "...                                                     ...  ..  \n",
       "Homo sapiens caspase recruitment domain family ...        0   1  \n",
       "Homo sapiens heterogeneous nuclear ribonucleopr...        0   1  \n",
       "Homo sapiens tumor protein p53 (TP53), transcri...        0   1  \n",
       "Homo sapiens sterile alpha motif domain contain...        0   1  \n",
       "Homo sapiens sterile alpha motif domain contain...        0   1  \n",
       "\n",
       "[12998 rows x 26 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df['sum'] = archea_df['TAA'] + archea_df['TGA'] + archea_df['TAG']\n",
    "indexNames = archea_df[ archea_df['sum'] == 0 ].index\n",
    "archea_df.drop(indexNames , inplace=True)\n",
    "\n",
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "archea_df.to_csv(r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Extracted_csvs\\human_8_gene_csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "def file_runner(data):\n",
    "    j = 0\n",
    "    codon_data = [['TAA',0],['TAG',0],['TGA',0],['unk',0]]\n",
    "    codon_unk =[]\n",
    "    codon_GC_data = [['TAA',0],['TAG',0],['TGA',0]]\n",
    "    codon_GC3_data = [['TAA',0],['TAG',0],['TGA',0]]\n",
    "    global_gene = code_rev(data,0)\n",
    "    global_GC = (global_gene.count('G')+global_gene.count('C'))/len(global_gene)\n",
    "    global_GC3 = GC_3(global_gene)\n",
    "    codon_four = [['TAAA',0],['TAAT',0],['TAAG',0],['TAAC',0],['TGAA',0],['TGAT',0],['TGAG',0],['TGAC',0],['TAGA',0],['TAGT',0],['TAGG',0],['TAGC',0],['unk',0]]\n",
    "    while j < len(data.features):\n",
    "        try:\n",
    "            if data.features[j].type == 'CDS':\n",
    "                gene = code_rev(data,j)\n",
    "                codon = gene[-3:]\n",
    "                GC_per = (gene.count('G')+gene.count('C'))/len(gene)\n",
    "                GC_3per = GC_3(gene)\n",
    "                if codon == 'TAA':\n",
    "                    codon_data[0][1] = codon_data[0][1]+1\n",
    "                    del(codon_GC_data[0][1])\n",
    "                    codon_GC_data[0].append(GC_per)\n",
    "                    del(codon_GC3_data[0][1])\n",
    "                    codon_GC3_data[0].append(GC_3per)\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                elif codon == 'TAG':\n",
    "                    codon_data[1][1] = codon_data[1][1]+1\n",
    "                    del(codon_GC_data[1][1])\n",
    "                    codon_GC_data[1].append(GC_per)\n",
    "                    del(codon_GC3_data[1][1])\n",
    "                    codon_GC3_data[1].append(GC_3per)\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                elif codon == 'TGA':\n",
    "                    codon_data[2][1] = codon_data[2][1]+1\n",
    "                    del(codon_GC_data[2][1])\n",
    "                    codon_GC_data[2].append(GC_per)\n",
    "                    del(codon_GC3_data[2][1])\n",
    "                    codon_GC3_data[2].append(GC_3per)\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                else:\n",
    "                    codon_data[3][1] = codon_data[3][1]+1\n",
    "                    codon_four = four_stop(data,j,codon_four)\n",
    "                    j+=1\n",
    "                    if len(codon_unk) ==0:\n",
    "                        codon_unk.append(data.description)\n",
    "                        codon_unk.append(codon)\n",
    "                    else:\n",
    "                        codon_unk.append(codon)\n",
    "                    \n",
    "            else:\n",
    "                j+=1\n",
    "        except:\n",
    "            j+=1\n",
    "    codon_df = pd.DataFrame(codon_GC_data, index = ['TAA','TAG','TGA'])\n",
    "    codon_GC3df = pd.DataFrame(codon_GC3_data, index = ['TAA','TAG','TGA'])\n",
    "    GC_list = (codon_df.mean(axis = 1))\n",
    "    GC3_list = (codon_GC3df.mean(axis=1))\n",
    "    Name_list_GC = ['Global_GC','TAA','TGA','TAG','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "    Name_list_GC3 = ['Global_GC3','TAA','TGA','TAG','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "    file_data_GC = [(global_GC*100),codon_data[0][1],codon_data[2][1],codon_data[1][1],codon_four[0][1],codon_four[1][1],codon_four[2][1],codon_four[3][1],codon_four[4][1],codon_four[5][1],codon_four[6][1],codon_four[7][1],codon_four[8][1],codon_four[9][1],codon_four[10][1],codon_four[11][1],codon_data[3][1],codon_four[12][1]]\n",
    "    file_data_GC3 = [(global_GC3*100),codon_data[0][1],codon_data[2][1],codon_data[1][1],codon_four[0][1],codon_four[1][1],codon_four[2][1],codon_four[3][1],codon_four[4][1],codon_four[5][1],codon_four[6][1],codon_four[7][1],codon_four[8][1],codon_four[9][1],codon_four[10][1],codon_four[11][1],codon_data[3][1],codon_four[12][1]]\n",
    "    return([file_data_GC,file_data_GC3])\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[43.6075036075036, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "holder = file_runner(record)\n",
    "tester_data = holder[0]\n",
    "tester_data\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Name_list_GC = ['Global_GC','TAA','TGA','TAG','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "len(Name_list_GC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TAAT</th>\n",
       "      <th>TAAG</th>\n",
       "      <th>TAAC</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global_GC</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TAA, TGA, TAG, TAAA, TAAT, TAAG, TAAC, TGAA, TGAT, TGAG, TGAC, TAGA, TAGT, TAGG, TAGC, Unk_Stop, Unk_four]\n",
       "Index: []"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Name_list = ['Global_GC','TAA','TGA','TAG','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "Homo_GC_data = pd.DataFrame(columns = Name_list)\n",
    "Homo_GC_data = Homo_GC_data.set_index('Global_GC')\n",
    "Homo_GC_data\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "Name_list_GC3 = ['Global_GC3','TAA','TGA','TAG','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TAAT</th>\n",
       "      <th>TAAG</th>\n",
       "      <th>TAAC</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global_GC3</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TAA, TGA, TAG, TAAA, TAAT, TAAG, TAAC, TGAA, TGAT, TGAG, TGAC, TAGA, TAGT, TAGG, TAGC, Unk_Stop, Unk_four]\n",
       "Index: []"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Homo_GC3_data = pd.DataFrame(columns = Name_list_GC3)\n",
    "Homo_GC3_data = Homo_GC3_data.set_index('Global_GC3')\n",
    "Homo_GC3_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "def file_adder(GC_data,new_data):\n",
    "    i = 0\n",
    "    while i < 10:\n",
    "        if new_data[0] < GC_data[i][0]:\n",
    "            x = (list(map(add, GC_data[i][1:], new_data[1:])))\n",
    "            x.insert(0,GC_full[i][0])\n",
    "            del(GC_data[i])\n",
    "            GC_data.insert(i,x)\n",
    "            return(GC_data)\n",
    "        else:\n",
    "            i+=1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "unk_stops = []\n",
    "GC_full = [[10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[30,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[40,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[50,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[60,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[70,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[80,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[90,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]\n",
    "GC3_full = [[10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[30,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[40,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[50,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[60,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[70,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[80,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[90,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]\n",
    "gb_file = r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Data\\Human_RNA\\human_8_rna.gbff'\n",
    "mega_file = SeqIO.parse(open(gb_file,\"r\"), \"genbank\")\n",
    "for record in mega_file:\n",
    "    if i <100:\n",
    "        holder = file_runner(record)\n",
    "        fresh_data = holder[0]\n",
    "        GC_full = file_adder(GC_full,fresh_data)\n",
    "        fre3sh_data = holder[1]\n",
    "        GC3_full = file_adder(GC3_full,fre3sh_data)\n",
    "        i+=1\n",
    "    else:\n",
    "        z= 0\n",
    "        while z < len(GC_full):\n",
    "            df_file_data = pd.DataFrame([GC_full[z]],columns = Name_list)\n",
    "            df_file_data = df_file_data.set_index('Global_GC')\n",
    "            Homo_GC_data = Homo_GC_data.append(df_file_data)\n",
    "                        \n",
    "            df_file_data3 = pd.DataFrame([GC3_full[z]], columns = Name_list_GC3)\n",
    "            df_file_data3 = df_file_data3.set_index('Global_GC3')\n",
    "            Homo_GC3_data = Homo_GC3_data.append(df_file_data3)\n",
    "            \n",
    "            z+=1\n",
    "        break\n",
    "z= 0\n",
    "while z < len(GC_full):\n",
    "    df_file_data = pd.DataFrame([GC_full[z]],columns = Name_list)\n",
    "    df_file_data = df_file_data.set_index('Global_GC')\n",
    "    Homo_GC_data = Homo_GC_data.append(df_file_data)\n",
    "\n",
    "    df_file_data3 = pd.DataFrame([GC3_full[z]], columns = Name_list_GC3)\n",
    "    df_file_data3 = df_file_data3.set_index('Global_GC3')\n",
    "    Homo_GC3_data = Homo_GC3_data.append(df_file_data3)\n",
    "\n",
    "    z+=1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TAAT</th>\n",
       "      <th>TAAG</th>\n",
       "      <th>TAAC</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global_GC</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>16</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>16</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          TAA TGA TAG TAAA TAAT TAAG TAAC TGAA TGAT TGAG TGAC TAGA TAGT TAGG  \\\n",
       "Global_GC                                                                      \n",
       "10          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "20          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "30          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "40          4  16   4    1    0    2    1   13    1    2    0    0    0    3   \n",
       "50         16  27   7    0    3    1   12   16    1    6    4    2    0    5   \n",
       "60          0   7   0    0    0    0    0    0    2    5    0    0    0    0   \n",
       "70          0   3   1    0    0    0    0    0    0    1    2    1    0    0   \n",
       "80          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "90          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "100         0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "10          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "20          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "30          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "40          4  16   4    1    0    2    1   13    1    2    0    0    0    3   \n",
       "50         16  27   7    0    3    1   12   16    1    6    4    2    0    5   \n",
       "60          0   7   0    0    0    0    0    0    2    5    0    0    0    0   \n",
       "70          0   3   1    0    0    0    0    0    0    1    2    1    0    0   \n",
       "80          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "90          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "100         0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "\n",
       "          TAGC Unk_Stop Unk_four  \n",
       "Global_GC                         \n",
       "10           0        0        0  \n",
       "20           0        0        0  \n",
       "30           0        0        0  \n",
       "40           1        0        0  \n",
       "50           0        0        0  \n",
       "60           0        0        0  \n",
       "70           0        0        0  \n",
       "80           0        0        0  \n",
       "90           0        0        0  \n",
       "100          0        0        0  \n",
       "10           0        0        0  \n",
       "20           0        0        0  \n",
       "30           0        0        0  \n",
       "40           1        0        0  \n",
       "50           0        0        0  \n",
       "60           0        0        0  \n",
       "70           0        0        0  \n",
       "80           0        0        0  \n",
       "90           0        0        0  \n",
       "100          0        0        0  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Homo_GC_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>TAAA</th>\n",
       "      <th>TAAT</th>\n",
       "      <th>TAAG</th>\n",
       "      <th>TAAC</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Global_GC3</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>14</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>14</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           TAA TGA TAG TAAA TAAT TAAG TAAC TGAA TGAT TGAG TGAC TAGA TAGT TAGG  \\\n",
       "Global_GC3                                                                      \n",
       "10           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "20           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "30           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "40          14  19   7    1    1    2   10   13    1    5    0    2    0    5   \n",
       "50           4  26   3    0    2    1    1   16    2    4    4    1    0    1   \n",
       "60           2   7   2    0    0    0    2    0    1    5    1    0    0    2   \n",
       "70           0   1   0    0    0    0    0    0    0    0    1    0    0    0   \n",
       "80           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "90           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "100          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "10           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "20           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "30           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "40          14  19   7    1    1    2   10   13    1    5    0    2    0    5   \n",
       "50           4  26   3    0    2    1    1   16    2    4    4    1    0    1   \n",
       "60           2   7   2    0    0    0    2    0    1    5    1    0    0    2   \n",
       "70           0   1   0    0    0    0    0    0    0    0    1    0    0    0   \n",
       "80           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "90           0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "100          0   0   0    0    0    0    0    0    0    0    0    0    0    0   \n",
       "\n",
       "           TAGC Unk_Stop Unk_four  \n",
       "Global_GC3                         \n",
       "10            0        0        0  \n",
       "20            0        0        0  \n",
       "30            0        0        0  \n",
       "40            0        0        0  \n",
       "50            1        0        0  \n",
       "60            0        0        0  \n",
       "70            0        0        0  \n",
       "80            0        0        0  \n",
       "90            0        0        0  \n",
       "100           0        0        0  \n",
       "10            0        0        0  \n",
       "20            0        0        0  \n",
       "30            0        0        0  \n",
       "40            0        0        0  \n",
       "50            1        0        0  \n",
       "60            0        0        0  \n",
       "70            0        0        0  \n",
       "80            0        0        0  \n",
       "90            0        0        0  \n",
       "100           0        0        0  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Homo_GC3_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "Homo_GC_data.to_csv(r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Extracted_csvs\\human_8_GC_csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "Homo_GC3_data.to_csv(r'C:\\Users\\Mole\\Documents\\Uni work\\UNI WORK\\Year 4\\The project\\Extracted_csvs\\human_8_GC3_csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "25"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GC_full = [['1-10',0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[8,8,8,8,8]]\n",
    "eee = [9,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]\n",
    "len(eee)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [],
   "source": [
    "Name_list = ['GC','TAA','TGA','TAG','Global_GC','Global_GC3','TAA_GC','TGA_GC','TAG_GC','TAA_GC3','TGA_GC3','TAG_GC3','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "Name_list3 = ['GC','TAA','TGA','TAG','Global_GC','Global_GC3','TAA_GC','TGA_GC','TAG_GC','TAA_GC3','TGA_GC3','TAG_GC3','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GC</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TAA, TGA, TAG, Global_GC, Global_GC3, TAA_GC, TGA_GC, TAG_GC, TAA_GC3, TGA_GC3, TAG_GC3, TAAA, TAAT, TAAG, TAAC, TGAA, TGAT, TGAG, TGAC, TAGA, TAGT, TAGG, TAGC, Unk_Stop, Unk_four]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 25 columns]"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Name_list = ['GC','TAA','TGA','TAG','Global_GC','Global_GC3','TAA_GC','TGA_GC','TAG_GC','TAA_GC3','TGA_GC3','TAG_GC3','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "archea_df = pd.DataFrame(columns = Name_list)\n",
    "archea_df = archea_df.set_index('GC')\n",
    "archea_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"None of ['GC'] are in the columns\"",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-225-57827e845432>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mName_list3\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m'AA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'Global_GC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'Global_GC3'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAA_GC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGA_GC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAG_GC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAA_GC3'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGA_GC3'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAG_GC3'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAAA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAAT'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAAG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAAC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGAA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGAT'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGAG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TGAC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAGA'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAGT'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAGG'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'TAGC'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'Unk_Stop'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'Unk_four'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0marchea_df3\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mName_list\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[0marchea_df3\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0marchea_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'GC'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m \u001b[0marchea_df3\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36mset_index\u001b[1;34m(self, keys, drop, append, inplace, verify_integrity)\u001b[0m\n\u001b[0;32m   4301\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4302\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mmissing\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 4303\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"None of {missing} are in the columns\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   4304\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4305\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0minplace\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: \"None of ['GC'] are in the columns\""
     ]
    }
   ],
   "source": [
    "Name_list3 = ['AA','TAA','TGA','TAG','Global_GC','Global_GC3','TAA_GC','TGA_GC','TAG_GC','TAA_GC3','TGA_GC3','TAG_GC3','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "archea_df3 = pd.DataFrame(columns = Name_list)\n",
    "archea_df3 = archea_df.set_index('GC')\n",
    "archea_df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [30, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [40, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [50, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [60, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [70, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0], [80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [90, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [100, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GC_full = [[10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[30,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[40,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[50,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[60,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[70,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[80,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[90,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]\n",
    "x = file_adder(GC_full,tester_data)\n",
    "print(x)\n",
    "len(x[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17"
      ]
     },
     "execution_count": 264,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Name_list_GC = ['Global_GC','TAA','TGA','TAG','TAAA','TAAT','TAAG','TAAC','TGAA','TGAT','TGAG','TGAC','TAGA','TAGT','TAGG','TAGC','Unk_Stop','Unk_four']\n",
    "len(Name_list_GC)\n",
    "file_data_GC = [(global_GC*100),codon_data[2][1],codon_data[1][1],codon_four[0][1],codon_four[1][1],codon_four[2][1],codon_four[3][1],codon_four[4][1],codon_four[5][1],codon_four[6][1],codon_four[7][1],codon_four[8][1],codon_four[9][1],codon_four[10][1],codon_four[11][1],codon_data[3][1],codon_four[12][1]]\n",
    "len(file_data_GC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[10, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(5, 0)"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = (list( map(add, GC_full[0][1:], eee[1:]) ))\n",
    "x.insert(0,GC_full[0][0])\n",
    "print(x)\n",
    "x = divmod(x[0],2)\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[1, 2, 3, 4], [10, 10, 10], [1, 2, 3, 4]]"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x = [[1,2,3,4],[1,2,3,4]]\n",
    "x.insert(1,[10,10,10])\n",
    "x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.0"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myArray = np.array([10,20,30,40,50,60,70,80,90])\n",
    "myInt = 10\n",
    "newArray = myArray myArray[0]/myInt\n",
    "newArray"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19"
      ]
     },
     "execution_count": 280,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GC_full = [[10,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[20,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[30,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[40,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[50,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[60,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[70,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[80,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[90,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]\n",
    "len(GC_full[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = [[0,0,0,1,2,3],[1,0,0,4,5,6,],[0,0,1,5,6,7]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0  1  2  3  4  5\n",
       "0  0  0  0  1  2  3\n",
       "1  1  0  0  4  5  6\n",
       "2  0  0  1  5  6  7"
      ]
     },
     "execution_count": 357,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ee = pd.DataFrame(df)\n",
    "ee"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 358,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>e</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0  1  2  3  4  5  e\n",
       "0  0  0  0  1  2  3  0\n",
       "1  1  0  0  4  5  6  1\n",
       "2  0  0  1  5  6  7  1"
      ]
     },
     "execution_count": 358,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ee['e'] = ee[0] + ee[1] + ee[2]\n",
    "ee"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n"
     ]
    }
   ],
   "source": [
    "ee\n",
    "x = sum(ee[0] + ee[1] + ee[2])\n",
    "print(x)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 360,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>e</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   0  1  2  3  4  5  e\n",
       "1  1  0  0  4  5  6  1\n",
       "2  0  0  1  5  6  7  1"
      ]
     },
     "execution_count": 360,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indexNames = ee[ ee['e'] == 0 ].index\n",
    "ee.drop(indexNames , inplace=True)\n",
    "\n",
    "ee"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 328,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAA</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 26, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.584505</td>\n",
       "      <td>0.563593</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.643661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.773006</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 27, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.588255</td>\n",
       "      <td>0.521654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.788703</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 32, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.360846</td>\n",
       "      <td>0.368810</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 38, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.360967</td>\n",
       "      <td>0.369171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens SWI/SNF related, matrix associated, actin dependent regulator of chromatin, subfamily a, member 4 (SMARCA4), transcript variant 8, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.604384</td>\n",
       "      <td>0.427945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.611558</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.632198</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens methyl-CpG binding protein 2 (MECP2), transcript variant 5, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.521316</td>\n",
       "      <td>0.523490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.604061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.588832</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens coagulation factor VIII (F8), transcript variant 2, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.424694</td>\n",
       "      <td>0.457569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.465438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.488479</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens potassium channel tetramerization domain containing 7 (KCTD7), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.501335</td>\n",
       "      <td>0.528651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.601149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.648276</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens luteinizing hormone subunit beta (LHB), mRNA</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.643130</td>\n",
       "      <td>0.761450</td>\n",
       "      <td>0.652582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633803</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens adenylate kinase 1 (AK1), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.578011</td>\n",
       "      <td>0.640511</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27381 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   TAA TGA TAG  Global_GC  \\\n",
       "Name                                                                        \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.584505   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.588255   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.360846   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.360967   \n",
       "Homo sapiens SWI/SNF related, matrix associated...   0   1   0   0.604384   \n",
       "...                                                 ..  ..  ..        ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...   0   1   0   0.521316   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...   0   1   0   0.424694   \n",
       "Homo sapiens potassium channel tetramerization ...   0   1   0   0.501335   \n",
       "Homo sapiens luteinizing hormone subunit beta (...   1   0   0   0.643130   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...   0   0   1   0.578011   \n",
       "\n",
       "                                                    Global_GC3    TAA_GC  \\\n",
       "Name                                                                       \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.563593       NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.521654       NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.368810       NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.369171       NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0.427945       NaN   \n",
       "...                                                        ...       ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...    0.523490       NaN   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...    0.457569       NaN   \n",
       "Homo sapiens potassium channel tetramerization ...    0.528651       NaN   \n",
       "Homo sapiens luteinizing hormone subunit beta (...    0.761450  0.652582   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...    0.640511       NaN   \n",
       "\n",
       "                                                      TGA_GC   TAG_GC  \\\n",
       "Name                                                                    \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.643661      NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.633194      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN      NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...  0.611558      NaN   \n",
       "...                                                      ...      ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...  0.604061      NaN   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...  0.465438      NaN   \n",
       "Homo sapiens potassium channel tetramerization ...  0.601149      NaN   \n",
       "Homo sapiens luteinizing hormone subunit beta (...       NaN      NaN   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...       NaN  0.57265   \n",
       "\n",
       "                                                     TAA_GC3   TGA_GC3  ...  \\\n",
       "Name                                                                    ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.773006  ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.788703  ...   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN       NaN  ...   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN       NaN  ...   \n",
       "Homo sapiens SWI/SNF related, matrix associated...       NaN  0.632198  ...   \n",
       "...                                                      ...       ...  ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...       NaN  0.588832  ...   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...       NaN  0.488479  ...   \n",
       "Homo sapiens potassium channel tetramerization ...       NaN  0.648276  ...   \n",
       "Homo sapiens luteinizing hormone subunit beta (...  0.633803       NaN  ...   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...       NaN       NaN  ...   \n",
       "\n",
       "                                                    TGAA TGAT TGAG TGAC TAGA  \\\n",
       "Name                                                                           \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     0    1    0    0    0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     0    1    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...     0    0    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...     0    0    0    0    0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...     0    0    1    0    0   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...     0    0    0    1    0   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...     0    0    1    0    0   \n",
       "Homo sapiens potassium channel tetramerization ...     0    0    1    0    0   \n",
       "Homo sapiens luteinizing hormone subunit beta (...     0    0    0    0    0   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...     0    0    0    0    0   \n",
       "\n",
       "                                                   TAGT TAGG TAGC Unk_Stop  \\\n",
       "Name                                                                         \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0    0        0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0    0        0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0    0        0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0    0        0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0    0    0        0   \n",
       "...                                                 ...  ...  ...      ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...    0    0    0        0   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...    0    0    0        0   \n",
       "Homo sapiens potassium channel tetramerization ...    0    0    0        0   \n",
       "Homo sapiens luteinizing hormone subunit beta (...    0    0    0        0   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...    0    0    1        0   \n",
       "\n",
       "                                                   Unk_four  \n",
       "Name                                                         \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0  \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0  \n",
       "Homo sapiens transforming acidic coiled-coil co...        0  \n",
       "Homo sapiens transforming acidic coiled-coil co...        0  \n",
       "Homo sapiens SWI/SNF related, matrix associated...        0  \n",
       "...                                                     ...  \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...        0  \n",
       "Homo sapiens coagulation factor VIII (F8), tran...        0  \n",
       "Homo sapiens potassium channel tetramerization ...        0  \n",
       "Homo sapiens luteinizing hormone subunit beta (...        0  \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...        0  \n",
       "\n",
       "[27381 rows x 25 columns]"
      ]
     },
     "execution_count": 328,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "      <th>sum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 26, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.584505</td>\n",
       "      <td>0.563593</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.643661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.773006</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 27, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.588255</td>\n",
       "      <td>0.521654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.788703</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 32, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.360846</td>\n",
       "      <td>0.368810</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens transforming acidic coiled-coil containing protein 1 (TACC1), transcript variant 38, non-coding RNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.360967</td>\n",
       "      <td>0.369171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens SWI/SNF related, matrix associated, actin dependent regulator of chromatin, subfamily a, member 4 (SMARCA4), transcript variant 8, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.604384</td>\n",
       "      <td>0.427945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.611558</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.632198</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens methyl-CpG binding protein 2 (MECP2), transcript variant 5, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.521316</td>\n",
       "      <td>0.523490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.604061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.588832</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens coagulation factor VIII (F8), transcript variant 2, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.424694</td>\n",
       "      <td>0.457569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.465438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.488479</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens potassium channel tetramerization domain containing 7 (KCTD7), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.501335</td>\n",
       "      <td>0.528651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.601149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.648276</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens luteinizing hormone subunit beta (LHB), mRNA</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.643130</td>\n",
       "      <td>0.761450</td>\n",
       "      <td>0.652582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633803</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens adenylate kinase 1 (AK1), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.578011</td>\n",
       "      <td>0.640511</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27381 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   TAA TGA TAG  Global_GC  \\\n",
       "Name                                                                        \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.584505   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.588255   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.360846   \n",
       "Homo sapiens transforming acidic coiled-coil co...   0   0   0   0.360967   \n",
       "Homo sapiens SWI/SNF related, matrix associated...   0   1   0   0.604384   \n",
       "...                                                 ..  ..  ..        ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...   0   1   0   0.521316   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...   0   1   0   0.424694   \n",
       "Homo sapiens potassium channel tetramerization ...   0   1   0   0.501335   \n",
       "Homo sapiens luteinizing hormone subunit beta (...   1   0   0   0.643130   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...   0   0   1   0.578011   \n",
       "\n",
       "                                                    Global_GC3    TAA_GC  \\\n",
       "Name                                                                       \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.563593       NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.521654       NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.368810       NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0.369171       NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0.427945       NaN   \n",
       "...                                                        ...       ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...    0.523490       NaN   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...    0.457569       NaN   \n",
       "Homo sapiens potassium channel tetramerization ...    0.528651       NaN   \n",
       "Homo sapiens luteinizing hormone subunit beta (...    0.761450  0.652582   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...    0.640511       NaN   \n",
       "\n",
       "                                                      TGA_GC   TAG_GC  \\\n",
       "Name                                                                    \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.643661      NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.633194      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN      NaN   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN      NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...  0.611558      NaN   \n",
       "...                                                      ...      ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...  0.604061      NaN   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...  0.465438      NaN   \n",
       "Homo sapiens potassium channel tetramerization ...  0.601149      NaN   \n",
       "Homo sapiens luteinizing hormone subunit beta (...       NaN      NaN   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...       NaN  0.57265   \n",
       "\n",
       "                                                     TAA_GC3   TGA_GC3  ...  \\\n",
       "Name                                                                    ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.773006  ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.788703  ...   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN       NaN  ...   \n",
       "Homo sapiens transforming acidic coiled-coil co...       NaN       NaN  ...   \n",
       "Homo sapiens SWI/SNF related, matrix associated...       NaN  0.632198  ...   \n",
       "...                                                      ...       ...  ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...       NaN  0.588832  ...   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...       NaN  0.488479  ...   \n",
       "Homo sapiens potassium channel tetramerization ...       NaN  0.648276  ...   \n",
       "Homo sapiens luteinizing hormone subunit beta (...  0.633803       NaN  ...   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...       NaN       NaN  ...   \n",
       "\n",
       "                                                    TGAT TGAG TGAC TAGA TAGT  \\\n",
       "Name                                                                           \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     1    0    0    0    0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     1    0    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...     0    0    0    0    0   \n",
       "Homo sapiens transforming acidic coiled-coil co...     0    0    0    0    0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...     0    1    0    0    0   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...     0    0    1    0    0   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...     0    1    0    0    0   \n",
       "Homo sapiens potassium channel tetramerization ...     0    1    0    0    0   \n",
       "Homo sapiens luteinizing hormone subunit beta (...     0    0    0    0    0   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...     0    0    0    0    0   \n",
       "\n",
       "                                                   TAGG TAGC Unk_Stop  \\\n",
       "Name                                                                    \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0        0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0        0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0        0   \n",
       "Homo sapiens transforming acidic coiled-coil co...    0    0        0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0    0        0   \n",
       "...                                                 ...  ...      ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...    0    0        0   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...    0    0        0   \n",
       "Homo sapiens potassium channel tetramerization ...    0    0        0   \n",
       "Homo sapiens luteinizing hormone subunit beta (...    0    0        0   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...    0    1        0   \n",
       "\n",
       "                                                   Unk_four sum  \n",
       "Name                                                             \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0   1  \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0   1  \n",
       "Homo sapiens transforming acidic coiled-coil co...        0   0  \n",
       "Homo sapiens transforming acidic coiled-coil co...        0   0  \n",
       "Homo sapiens SWI/SNF related, matrix associated...        0   1  \n",
       "...                                                     ...  ..  \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...        0   1  \n",
       "Homo sapiens coagulation factor VIII (F8), tran...        0   1  \n",
       "Homo sapiens potassium channel tetramerization ...        0   1  \n",
       "Homo sapiens luteinizing hormone subunit beta (...        0   1  \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...        0   1  \n",
       "\n",
       "[27381 rows x 26 columns]"
      ]
     },
     "execution_count": 362,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "archea_df['sum'] = archea_df['TAA'] + archea_df['TGA'] + archea_df['TAG']\n",
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TAA</th>\n",
       "      <th>TGA</th>\n",
       "      <th>TAG</th>\n",
       "      <th>Global_GC</th>\n",
       "      <th>Global_GC3</th>\n",
       "      <th>TAA_GC</th>\n",
       "      <th>TGA_GC</th>\n",
       "      <th>TAG_GC</th>\n",
       "      <th>TAA_GC3</th>\n",
       "      <th>TGA_GC3</th>\n",
       "      <th>...</th>\n",
       "      <th>TGAT</th>\n",
       "      <th>TGAG</th>\n",
       "      <th>TGAC</th>\n",
       "      <th>TAGA</th>\n",
       "      <th>TAGT</th>\n",
       "      <th>TAGG</th>\n",
       "      <th>TAGC</th>\n",
       "      <th>Unk_Stop</th>\n",
       "      <th>Unk_four</th>\n",
       "      <th>sum</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 26, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.584505</td>\n",
       "      <td>0.563593</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.643661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.773006</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 27, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.588255</td>\n",
       "      <td>0.521654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.788703</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens SWI/SNF related, matrix associated, actin dependent regulator of chromatin, subfamily a, member 4 (SMARCA4), transcript variant 8, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.604384</td>\n",
       "      <td>0.427945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.611558</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.632198</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens small integral membrane protein 35 (SMIM35), mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.566416</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.557471</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.568966</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens collagen type XIII alpha 1 chain (COL13A1), transcript variant 26, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.584505</td>\n",
       "      <td>0.563593</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.643661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.773006</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens methyl-CpG binding protein 2 (MECP2), transcript variant 5, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.521316</td>\n",
       "      <td>0.523490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.604061</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.588832</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens coagulation factor VIII (F8), transcript variant 2, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.424694</td>\n",
       "      <td>0.457569</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.465438</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.488479</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens potassium channel tetramerization domain containing 7 (KCTD7), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.501335</td>\n",
       "      <td>0.528651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.601149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.648276</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens luteinizing hormone subunit beta (LHB), mRNA</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.643130</td>\n",
       "      <td>0.761450</td>\n",
       "      <td>0.652582</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.633803</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Homo sapiens adenylate kinase 1 (AK1), transcript variant 1, mRNA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.578011</td>\n",
       "      <td>0.640511</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57265</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11350 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   TAA TGA TAG  Global_GC  \\\n",
       "Name                                                                        \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.584505   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.588255   \n",
       "Homo sapiens SWI/SNF related, matrix associated...   0   1   0   0.604384   \n",
       "Homo sapiens small integral membrane protein 35...   0   1   0   0.566416   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...   0   1   0   0.584505   \n",
       "...                                                 ..  ..  ..        ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...   0   1   0   0.521316   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...   0   1   0   0.424694   \n",
       "Homo sapiens potassium channel tetramerization ...   0   1   0   0.501335   \n",
       "Homo sapiens luteinizing hormone subunit beta (...   1   0   0   0.643130   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...   0   0   1   0.578011   \n",
       "\n",
       "                                                    Global_GC3    TAA_GC  \\\n",
       "Name                                                                       \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.563593       NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.521654       NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0.427945       NaN   \n",
       "Homo sapiens small integral membrane protein 35...    0.571429       NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0.563593       NaN   \n",
       "...                                                        ...       ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...    0.523490       NaN   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...    0.457569       NaN   \n",
       "Homo sapiens potassium channel tetramerization ...    0.528651       NaN   \n",
       "Homo sapiens luteinizing hormone subunit beta (...    0.761450  0.652582   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...    0.640511       NaN   \n",
       "\n",
       "                                                      TGA_GC   TAG_GC  \\\n",
       "Name                                                                    \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.643661      NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.633194      NaN   \n",
       "Homo sapiens SWI/SNF related, matrix associated...  0.611558      NaN   \n",
       "Homo sapiens small integral membrane protein 35...  0.557471      NaN   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...  0.643661      NaN   \n",
       "...                                                      ...      ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...  0.604061      NaN   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...  0.465438      NaN   \n",
       "Homo sapiens potassium channel tetramerization ...  0.601149      NaN   \n",
       "Homo sapiens luteinizing hormone subunit beta (...       NaN      NaN   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...       NaN  0.57265   \n",
       "\n",
       "                                                     TAA_GC3   TGA_GC3  ...  \\\n",
       "Name                                                                    ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.773006  ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.788703  ...   \n",
       "Homo sapiens SWI/SNF related, matrix associated...       NaN  0.632198  ...   \n",
       "Homo sapiens small integral membrane protein 35...       NaN  0.568966  ...   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...       NaN  0.773006  ...   \n",
       "...                                                      ...       ...  ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...       NaN  0.588832  ...   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...       NaN  0.488479  ...   \n",
       "Homo sapiens potassium channel tetramerization ...       NaN  0.648276  ...   \n",
       "Homo sapiens luteinizing hormone subunit beta (...  0.633803       NaN  ...   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...       NaN       NaN  ...   \n",
       "\n",
       "                                                    TGAT TGAG TGAC TAGA TAGT  \\\n",
       "Name                                                                           \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     1    0    0    0    0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     1    0    0    0    0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...     0    1    0    0    0   \n",
       "Homo sapiens small integral membrane protein 35...     1    0    0    0    0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...     1    0    0    0    0   \n",
       "...                                                  ...  ...  ...  ...  ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...     0    0    1    0    0   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...     0    1    0    0    0   \n",
       "Homo sapiens potassium channel tetramerization ...     0    1    0    0    0   \n",
       "Homo sapiens luteinizing hormone subunit beta (...     0    0    0    0    0   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...     0    0    0    0    0   \n",
       "\n",
       "                                                   TAGG TAGC Unk_Stop  \\\n",
       "Name                                                                    \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0        0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0        0   \n",
       "Homo sapiens SWI/SNF related, matrix associated...    0    0        0   \n",
       "Homo sapiens small integral membrane protein 35...    0    0        0   \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...    0    0        0   \n",
       "...                                                 ...  ...      ...   \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...    0    0        0   \n",
       "Homo sapiens coagulation factor VIII (F8), tran...    0    0        0   \n",
       "Homo sapiens potassium channel tetramerization ...    0    0        0   \n",
       "Homo sapiens luteinizing hormone subunit beta (...    0    0        0   \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...    0    1        0   \n",
       "\n",
       "                                                   Unk_four sum  \n",
       "Name                                                             \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0   1  \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0   1  \n",
       "Homo sapiens SWI/SNF related, matrix associated...        0   1  \n",
       "Homo sapiens small integral membrane protein 35...        0   1  \n",
       "Homo sapiens collagen type XIII alpha 1 chain (...        0   1  \n",
       "...                                                     ...  ..  \n",
       "Homo sapiens methyl-CpG binding protein 2 (MECP...        0   1  \n",
       "Homo sapiens coagulation factor VIII (F8), tran...        0   1  \n",
       "Homo sapiens potassium channel tetramerization ...        0   1  \n",
       "Homo sapiens luteinizing hormone subunit beta (...        0   1  \n",
       "Homo sapiens adenylate kinase 1 (AK1), transcri...        0   1  \n",
       "\n",
       "[11350 rows x 26 columns]"
      ]
     },
     "execution_count": 363,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indexNames = archea_df[ archea_df['sum'] == 0 ].index\n",
    "archea_df.drop(indexNames , inplace=True)\n",
    "\n",
    "archea_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "archea_df['sum'] = archea_df['TAA'] + archea_df['TGA'] + archea_df['TAG']\n",
    "indexNames = archea_df[ archea_df['sum'] == 0 ].index\n",
    "archea_df.drop(indexNames , inplace=True)\n",
    "\n",
    "archea_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
